// vite-plugin-pages-sitemap/dist/index.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __commonJS = (cb, mod) => function __require() {
  return mod || (0, cb[__getOwnPropNames(cb)[0]])((mod = { exports: {} }).exports, mod), mod.exports;
};
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var __accessCheck = (obj, member, msg) => {
  if (!member.has(obj))
    throw TypeError("Cannot " + msg);
};
var __privateGet = (obj, member, getter) => {
  __accessCheck(obj, member, "read from private field");
  return getter ? getter.call(obj) : member.get(obj);
};
var __privateAdd = (obj, member, value) => {
  if (member.has(obj))
    throw TypeError("Cannot add the same private member more than once");
  member instanceof WeakSet ? member.add(obj) : member.set(obj, value);
};
var __privateSet = (obj, member, value, setter) => {
  __accessCheck(obj, member, "write to private field");
  setter ? setter.call(obj, value) : member.set(obj, value);
  return value;
};
var __privateWrapper = (obj, member, setter, getter) => {
  return {
    set _(value) {
      __privateSet(obj, member, value, setter);
    },
    get _() {
      return __privateGet(obj, member, getter);
    }
  };
};

// node_modules/.pnpm/sitemap@7.1.1/node_modules/sitemap/dist/lib/errors.js
var require_errors = __commonJS({
  "node_modules/.pnpm/sitemap@7.1.1/node_modules/sitemap/dist/lib/errors.js"(exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.EmptySitemap = exports.EmptyStream = exports.InvalidVideoPriceCurrency = exports.InvalidVideoResolution = exports.InvalidVideoPriceType = exports.InvalidVideoRestrictionRelationship = exports.InvalidVideoRestriction = exports.InvalidVideoFamilyFriendly = exports.InvalidVideoCategory = exports.InvalidVideoTagCount = exports.InvalidVideoViewCount = exports.InvalidVideoTitle = exports.XMLLintUnavailable = exports.InvalidNewsAccessValue = exports.InvalidNewsFormat = exports.InvalidAttr = exports.InvalidAttrValue = exports.InvalidVideoRating = exports.InvalidVideoDescription = exports.InvalidVideoDuration = exports.InvalidVideoFormat = exports.UndefinedTargetFolder = exports.PriorityInvalidError = exports.ChangeFreqInvalidError = exports.NoConfigError = exports.NoURLError = void 0;
    var NoURLError = class extends Error {
      constructor(message) {
        super(message || "URL is required");
        this.name = "NoURLError";
        Error.captureStackTrace(this, NoURLError);
      }
    };
    exports.NoURLError = NoURLError;
    var NoConfigError = class extends Error {
      constructor(message) {
        super(message || "SitemapItem requires a configuration");
        this.name = "NoConfigError";
        Error.captureStackTrace(this, NoConfigError);
      }
    };
    exports.NoConfigError = NoConfigError;
    var ChangeFreqInvalidError = class extends Error {
      constructor(url, changefreq) {
        super(`${url}: changefreq "${changefreq}" is invalid`);
        this.name = "ChangeFreqInvalidError";
        Error.captureStackTrace(this, ChangeFreqInvalidError);
      }
    };
    exports.ChangeFreqInvalidError = ChangeFreqInvalidError;
    var PriorityInvalidError = class extends Error {
      constructor(url, priority) {
        super(`${url}: priority "${priority}" must be a number between 0 and 1 inclusive`);
        this.name = "PriorityInvalidError";
        Error.captureStackTrace(this, PriorityInvalidError);
      }
    };
    exports.PriorityInvalidError = PriorityInvalidError;
    var UndefinedTargetFolder = class extends Error {
      constructor(message) {
        super(message || "Target folder must exist");
        this.name = "UndefinedTargetFolder";
        Error.captureStackTrace(this, UndefinedTargetFolder);
      }
    };
    exports.UndefinedTargetFolder = UndefinedTargetFolder;
    var InvalidVideoFormat = class extends Error {
      constructor(url) {
        super(`${url} video must include thumbnail_loc, title and description fields for videos`);
        this.name = "InvalidVideoFormat";
        Error.captureStackTrace(this, InvalidVideoFormat);
      }
    };
    exports.InvalidVideoFormat = InvalidVideoFormat;
    var InvalidVideoDuration = class extends Error {
      constructor(url, duration) {
        super(`${url} duration "${duration}" must be an integer of seconds between 0 and 28800`);
        this.name = "InvalidVideoDuration";
        Error.captureStackTrace(this, InvalidVideoDuration);
      }
    };
    exports.InvalidVideoDuration = InvalidVideoDuration;
    var InvalidVideoDescription = class extends Error {
      constructor(url, length) {
        const message = `${url}: video description is too long ${length} vs limit of 2048 characters.`;
        super(message);
        this.name = "InvalidVideoDescription";
        Error.captureStackTrace(this, InvalidVideoDescription);
      }
    };
    exports.InvalidVideoDescription = InvalidVideoDescription;
    var InvalidVideoRating = class extends Error {
      constructor(url, title, rating) {
        super(`${url}: video "${title}" rating "${rating}" must be between 0 and 5 inclusive`);
        this.name = "InvalidVideoRating";
        Error.captureStackTrace(this, InvalidVideoRating);
      }
    };
    exports.InvalidVideoRating = InvalidVideoRating;
    var InvalidAttrValue = class extends Error {
      constructor(key, val, validator) {
        super('"' + val + '" tested against: ' + validator + ' is not a valid value for attr: "' + key + '"');
        this.name = "InvalidAttrValue";
        Error.captureStackTrace(this, InvalidAttrValue);
      }
    };
    exports.InvalidAttrValue = InvalidAttrValue;
    var InvalidAttr = class extends Error {
      constructor(key) {
        super('"' + key + '" is malformed');
        this.name = "InvalidAttr";
        Error.captureStackTrace(this, InvalidAttr);
      }
    };
    exports.InvalidAttr = InvalidAttr;
    var InvalidNewsFormat = class extends Error {
      constructor(url) {
        super(`${url} News must include publication, publication name, publication language, title, and publication_date for news`);
        this.name = "InvalidNewsFormat";
        Error.captureStackTrace(this, InvalidNewsFormat);
      }
    };
    exports.InvalidNewsFormat = InvalidNewsFormat;
    var InvalidNewsAccessValue = class extends Error {
      constructor(url, access) {
        super(`${url} News access "${access}" must be either Registration, Subscription or not be present`);
        this.name = "InvalidNewsAccessValue";
        Error.captureStackTrace(this, InvalidNewsAccessValue);
      }
    };
    exports.InvalidNewsAccessValue = InvalidNewsAccessValue;
    var XMLLintUnavailable = class extends Error {
      constructor(message) {
        super(message || "xmlLint is not installed. XMLLint is required to validate");
        this.name = "XMLLintUnavailable";
        Error.captureStackTrace(this, XMLLintUnavailable);
      }
    };
    exports.XMLLintUnavailable = XMLLintUnavailable;
    var InvalidVideoTitle = class extends Error {
      constructor(url, length) {
        super(`${url}: video title is too long ${length} vs 100 character limit`);
        this.name = "InvalidVideoTitle";
        Error.captureStackTrace(this, InvalidVideoTitle);
      }
    };
    exports.InvalidVideoTitle = InvalidVideoTitle;
    var InvalidVideoViewCount = class extends Error {
      constructor(url, count) {
        super(`${url}: video view count must be positive, view count was ${count}`);
        this.name = "InvalidVideoViewCount";
        Error.captureStackTrace(this, InvalidVideoViewCount);
      }
    };
    exports.InvalidVideoViewCount = InvalidVideoViewCount;
    var InvalidVideoTagCount = class extends Error {
      constructor(url, count) {
        super(`${url}: video can have no more than 32 tags, this has ${count}`);
        this.name = "InvalidVideoTagCount";
        Error.captureStackTrace(this, InvalidVideoTagCount);
      }
    };
    exports.InvalidVideoTagCount = InvalidVideoTagCount;
    var InvalidVideoCategory = class extends Error {
      constructor(url, count) {
        super(`${url}: video category can only be 256 characters but was passed ${count}`);
        this.name = "InvalidVideoCategory";
        Error.captureStackTrace(this, InvalidVideoCategory);
      }
    };
    exports.InvalidVideoCategory = InvalidVideoCategory;
    var InvalidVideoFamilyFriendly = class extends Error {
      constructor(url, fam) {
        super(`${url}: video family friendly must be yes or no, was passed "${fam}"`);
        this.name = "InvalidVideoFamilyFriendly";
        Error.captureStackTrace(this, InvalidVideoFamilyFriendly);
      }
    };
    exports.InvalidVideoFamilyFriendly = InvalidVideoFamilyFriendly;
    var InvalidVideoRestriction = class extends Error {
      constructor(url, code) {
        super(`${url}: video restriction must be one or more two letter country codes. Was passed "${code}"`);
        this.name = "InvalidVideoRestriction";
        Error.captureStackTrace(this, InvalidVideoRestriction);
      }
    };
    exports.InvalidVideoRestriction = InvalidVideoRestriction;
    var InvalidVideoRestrictionRelationship = class extends Error {
      constructor(url, val) {
        super(`${url}: video restriction relationship must be either allow or deny. Was passed "${val}"`);
        this.name = "InvalidVideoRestrictionRelationship";
        Error.captureStackTrace(this, InvalidVideoRestrictionRelationship);
      }
    };
    exports.InvalidVideoRestrictionRelationship = InvalidVideoRestrictionRelationship;
    var InvalidVideoPriceType = class extends Error {
      constructor(url, priceType, price) {
        super(priceType === void 0 && price === "" ? `${url}: video priceType is required when price is not provided` : `${url}: video price type "${priceType}" is not "rent" or "purchase"`);
        this.name = "InvalidVideoPriceType";
        Error.captureStackTrace(this, InvalidVideoPriceType);
      }
    };
    exports.InvalidVideoPriceType = InvalidVideoPriceType;
    var InvalidVideoResolution = class extends Error {
      constructor(url, resolution) {
        super(`${url}: video price resolution "${resolution}" is not hd or sd`);
        this.name = "InvalidVideoResolution";
        Error.captureStackTrace(this, InvalidVideoResolution);
      }
    };
    exports.InvalidVideoResolution = InvalidVideoResolution;
    var InvalidVideoPriceCurrency = class extends Error {
      constructor(url, currency) {
        super(`${url}: video price currency "${currency}" must be a three capital letter abbrieviation for the country currency`);
        this.name = "InvalidVideoPriceCurrency";
        Error.captureStackTrace(this, InvalidVideoPriceCurrency);
      }
    };
    exports.InvalidVideoPriceCurrency = InvalidVideoPriceCurrency;
    var EmptyStream = class extends Error {
      constructor() {
        super("You have ended the stream before anything was written. streamToPromise MUST be called before ending the stream.");
        this.name = "EmptyStream";
        Error.captureStackTrace(this, EmptyStream);
      }
    };
    exports.EmptyStream = EmptyStream;
    var EmptySitemap = class extends Error {
      constructor() {
        super("You ended the stream without writing anything.");
        this.name = "EmptySitemap";
        Error.captureStackTrace(this, EmptyStream);
      }
    };
    exports.EmptySitemap = EmptySitemap;
  }
});

// node_modules/.pnpm/sitemap@7.1.1/node_modules/sitemap/dist/lib/types.js
var require_types = __commonJS({
  "node_modules/.pnpm/sitemap@7.1.1/node_modules/sitemap/dist/lib/types.js"(exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.IndexTagNames = exports.TagNames = exports.ErrorLevel = exports.isAllowDeny = exports.EnumAllowDeny = exports.isValidYesNo = exports.EnumYesNo = exports.isValidChangeFreq = exports.CHANGEFREQ = exports.isResolution = exports.isPriceType = exports.validators = exports.EnumChangefreq = void 0;
    var EnumChangefreq;
    (function(EnumChangefreq2) {
      EnumChangefreq2["DAILY"] = "daily";
      EnumChangefreq2["MONTHLY"] = "monthly";
      EnumChangefreq2["ALWAYS"] = "always";
      EnumChangefreq2["HOURLY"] = "hourly";
      EnumChangefreq2["WEEKLY"] = "weekly";
      EnumChangefreq2["YEARLY"] = "yearly";
      EnumChangefreq2["NEVER"] = "never";
    })(EnumChangefreq = exports.EnumChangefreq || (exports.EnumChangefreq = {}));
    var allowDeny = /^(?:allow|deny)$/;
    exports.validators = {
      "price:currency": /^[A-Z]{3}$/,
      "price:type": /^(?:rent|purchase|RENT|PURCHASE)$/,
      "price:resolution": /^(?:HD|hd|sd|SD)$/,
      "platform:relationship": allowDeny,
      "restriction:relationship": allowDeny,
      restriction: /^([A-Z]{2}( +[A-Z]{2})*)?$/,
      platform: /^((web|mobile|tv)( (web|mobile|tv))*)?$/,
      language: /^zh-cn|zh-tw|([a-z]{2,3})$/,
      genres: /^(PressRelease|Satire|Blog|OpEd|Opinion|UserGenerated)(, *(PressRelease|Satire|Blog|OpEd|Opinion|UserGenerated))*$/,
      stock_tickers: /^(\w+:\w+(, *\w+:\w+){0,4})?$/
    };
    function isPriceType(pt) {
      return exports.validators["price:type"].test(pt);
    }
    exports.isPriceType = isPriceType;
    function isResolution(res) {
      return exports.validators["price:resolution"].test(res);
    }
    exports.isResolution = isResolution;
    exports.CHANGEFREQ = Object.values(EnumChangefreq);
    function isValidChangeFreq(freq) {
      return exports.CHANGEFREQ.includes(freq);
    }
    exports.isValidChangeFreq = isValidChangeFreq;
    var EnumYesNo;
    (function(EnumYesNo2) {
      EnumYesNo2["YES"] = "YES";
      EnumYesNo2["NO"] = "NO";
      EnumYesNo2["Yes"] = "Yes";
      EnumYesNo2["No"] = "No";
      EnumYesNo2["yes"] = "yes";
      EnumYesNo2["no"] = "no";
    })(EnumYesNo = exports.EnumYesNo || (exports.EnumYesNo = {}));
    function isValidYesNo(yn) {
      return /^YES|NO|[Yy]es|[Nn]o$/.test(yn);
    }
    exports.isValidYesNo = isValidYesNo;
    var EnumAllowDeny;
    (function(EnumAllowDeny2) {
      EnumAllowDeny2["ALLOW"] = "allow";
      EnumAllowDeny2["DENY"] = "deny";
    })(EnumAllowDeny = exports.EnumAllowDeny || (exports.EnumAllowDeny = {}));
    function isAllowDeny(ad) {
      return allowDeny.test(ad);
    }
    exports.isAllowDeny = isAllowDeny;
    var ErrorLevel;
    (function(ErrorLevel2) {
      ErrorLevel2["SILENT"] = "silent";
      ErrorLevel2["WARN"] = "warn";
      ErrorLevel2["THROW"] = "throw";
    })(ErrorLevel = exports.ErrorLevel || (exports.ErrorLevel = {}));
    var TagNames;
    (function(TagNames2) {
      TagNames2["url"] = "url";
      TagNames2["loc"] = "loc";
      TagNames2["urlset"] = "urlset";
      TagNames2["lastmod"] = "lastmod";
      TagNames2["changefreq"] = "changefreq";
      TagNames2["priority"] = "priority";
      TagNames2["video:thumbnail_loc"] = "video:thumbnail_loc";
      TagNames2["video:video"] = "video:video";
      TagNames2["video:title"] = "video:title";
      TagNames2["video:description"] = "video:description";
      TagNames2["video:tag"] = "video:tag";
      TagNames2["video:duration"] = "video:duration";
      TagNames2["video:player_loc"] = "video:player_loc";
      TagNames2["video:content_loc"] = "video:content_loc";
      TagNames2["image:image"] = "image:image";
      TagNames2["image:loc"] = "image:loc";
      TagNames2["image:geo_location"] = "image:geo_location";
      TagNames2["image:license"] = "image:license";
      TagNames2["image:title"] = "image:title";
      TagNames2["image:caption"] = "image:caption";
      TagNames2["video:requires_subscription"] = "video:requires_subscription";
      TagNames2["video:publication_date"] = "video:publication_date";
      TagNames2["video:id"] = "video:id";
      TagNames2["video:restriction"] = "video:restriction";
      TagNames2["video:family_friendly"] = "video:family_friendly";
      TagNames2["video:view_count"] = "video:view_count";
      TagNames2["video:uploader"] = "video:uploader";
      TagNames2["video:expiration_date"] = "video:expiration_date";
      TagNames2["video:platform"] = "video:platform";
      TagNames2["video:price"] = "video:price";
      TagNames2["video:rating"] = "video:rating";
      TagNames2["video:category"] = "video:category";
      TagNames2["video:live"] = "video:live";
      TagNames2["video:gallery_loc"] = "video:gallery_loc";
      TagNames2["news:news"] = "news:news";
      TagNames2["news:publication"] = "news:publication";
      TagNames2["news:name"] = "news:name";
      TagNames2["news:access"] = "news:access";
      TagNames2["news:genres"] = "news:genres";
      TagNames2["news:publication_date"] = "news:publication_date";
      TagNames2["news:title"] = "news:title";
      TagNames2["news:keywords"] = "news:keywords";
      TagNames2["news:stock_tickers"] = "news:stock_tickers";
      TagNames2["news:language"] = "news:language";
      TagNames2["mobile:mobile"] = "mobile:mobile";
      TagNames2["xhtml:link"] = "xhtml:link";
      TagNames2["expires"] = "expires";
    })(TagNames = exports.TagNames || (exports.TagNames = {}));
    var IndexTagNames;
    (function(IndexTagNames2) {
      IndexTagNames2["sitemap"] = "sitemap";
      IndexTagNames2["sitemapindex"] = "sitemapindex";
      IndexTagNames2["loc"] = "loc";
      IndexTagNames2["lastmod"] = "lastmod";
    })(IndexTagNames = exports.IndexTagNames || (exports.IndexTagNames = {}));
  }
});

// node_modules/.pnpm/sitemap@7.1.1/node_modules/sitemap/dist/lib/sitemap-xml.js
var require_sitemap_xml = __commonJS({
  "node_modules/.pnpm/sitemap@7.1.1/node_modules/sitemap/dist/lib/sitemap-xml.js"(exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.element = exports.ctag = exports.otag = exports.text = void 0;
    var invalidXMLUnicodeRegex = /[\u0000-\u0008\u000B\u000C\u000E-\u001F\u007F-\u0084\u0086-\u009F\uD800-\uDFFF\uFDD0-\uFDDF\u{1FFFE}-\u{1FFFF}\u{2FFFE}-\u{2FFFF}\u{3FFFE}-\u{3FFFF}\u{4FFFE}-\u{4FFFF}\u{5FFFE}-\u{5FFFF}\u{6FFFE}-\u{6FFFF}\u{7FFFE}-\u{7FFFF}\u{8FFFE}-\u{8FFFF}\u{9FFFE}-\u{9FFFF}\u{AFFFE}-\u{AFFFF}\u{BFFFE}-\u{BFFFF}\u{CFFFE}-\u{CFFFF}\u{DFFFE}-\u{DFFFF}\u{EFFFE}-\u{EFFFF}\u{FFFFE}-\u{FFFFF}\u{10FFFE}-\u{10FFFF}]/gu;
    var amp = /&/g;
    var lt = /</g;
    var apos = /'/g;
    var quot = /"/g;
    function text(txt) {
      return txt.replace(amp, "&amp;").replace(lt, "&lt;").replace(invalidXMLUnicodeRegex, "");
    }
    exports.text = text;
    function otag(nodeName, attrs, selfClose = false) {
      let attrstr = "";
      for (const k in attrs) {
        const val = attrs[k].replace(amp, "&amp;").replace(lt, "&lt;").replace(apos, "&apos;").replace(quot, "&quot;").replace(invalidXMLUnicodeRegex, "");
        attrstr += ` ${k}="${val}"`;
      }
      return `<${nodeName}${attrstr}${selfClose ? "/" : ""}>`;
    }
    exports.otag = otag;
    function ctag(nodeName) {
      return `</${nodeName}>`;
    }
    exports.ctag = ctag;
    function element(nodeName, attrs, innerText) {
      if (typeof attrs === "string") {
        return otag(nodeName) + text(attrs) + ctag(nodeName);
      } else if (innerText) {
        return otag(nodeName, attrs) + text(innerText) + ctag(nodeName);
      } else {
        return otag(nodeName, attrs, true);
      }
    }
    exports.element = element;
  }
});

// node_modules/.pnpm/sitemap@7.1.1/node_modules/sitemap/dist/lib/sitemap-item-stream.js
var require_sitemap_item_stream = __commonJS({
  "node_modules/.pnpm/sitemap@7.1.1/node_modules/sitemap/dist/lib/sitemap-item-stream.js"(exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.SitemapItemStream = void 0;
    var stream_1 = require("stream");
    var errors_1 = require_errors();
    var types_1 = require_types();
    var sitemap_xml_1 = require_sitemap_xml();
    function attrBuilder(conf, keys) {
      if (typeof keys === "string") {
        keys = [keys];
      }
      const iv = {};
      return keys.reduce((attrs, key) => {
        if (conf[key] !== void 0) {
          const keyAr = key.split(":");
          if (keyAr.length !== 2) {
            throw new errors_1.InvalidAttr(key);
          }
          attrs[keyAr[1]] = conf[key];
        }
        return attrs;
      }, iv);
    }
    var SitemapItemStream = class extends stream_1.Transform {
      constructor(opts = { level: types_1.ErrorLevel.WARN }) {
        opts.objectMode = true;
        super(opts);
        this.level = opts.level || types_1.ErrorLevel.WARN;
      }
      _transform(item, encoding, callback) {
        this.push((0, sitemap_xml_1.otag)(types_1.TagNames.url));
        this.push((0, sitemap_xml_1.element)(types_1.TagNames.loc, item.url));
        if (item.lastmod) {
          this.push((0, sitemap_xml_1.element)(types_1.TagNames.lastmod, item.lastmod));
        }
        if (item.changefreq) {
          this.push((0, sitemap_xml_1.element)(types_1.TagNames.changefreq, item.changefreq));
        }
        if (item.priority !== void 0 && item.priority !== null) {
          if (item.fullPrecisionPriority) {
            this.push((0, sitemap_xml_1.element)(types_1.TagNames.priority, item.priority.toString()));
          } else {
            this.push((0, sitemap_xml_1.element)(types_1.TagNames.priority, item.priority.toFixed(1)));
          }
        }
        item.video.forEach((video) => {
          this.push((0, sitemap_xml_1.otag)(types_1.TagNames["video:video"]));
          this.push((0, sitemap_xml_1.element)(types_1.TagNames["video:thumbnail_loc"], video.thumbnail_loc));
          this.push((0, sitemap_xml_1.element)(types_1.TagNames["video:title"], video.title));
          this.push((0, sitemap_xml_1.element)(types_1.TagNames["video:description"], video.description));
          if (video.content_loc) {
            this.push((0, sitemap_xml_1.element)(types_1.TagNames["video:content_loc"], video.content_loc));
          }
          if (video.player_loc) {
            this.push((0, sitemap_xml_1.element)(types_1.TagNames["video:player_loc"], attrBuilder(video, [
              "player_loc:autoplay",
              "player_loc:allow_embed"
            ]), video.player_loc));
          }
          if (video.duration) {
            this.push((0, sitemap_xml_1.element)(types_1.TagNames["video:duration"], video.duration.toString()));
          }
          if (video.expiration_date) {
            this.push((0, sitemap_xml_1.element)(types_1.TagNames["video:expiration_date"], video.expiration_date));
          }
          if (video.rating !== void 0) {
            this.push((0, sitemap_xml_1.element)(types_1.TagNames["video:rating"], video.rating.toString()));
          }
          if (video.view_count !== void 0) {
            this.push((0, sitemap_xml_1.element)(types_1.TagNames["video:view_count"], video.view_count.toString()));
          }
          if (video.publication_date) {
            this.push((0, sitemap_xml_1.element)(types_1.TagNames["video:publication_date"], video.publication_date));
          }
          for (const tag of video.tag) {
            this.push((0, sitemap_xml_1.element)(types_1.TagNames["video:tag"], tag));
          }
          if (video.category) {
            this.push((0, sitemap_xml_1.element)(types_1.TagNames["video:category"], video.category));
          }
          if (video.family_friendly) {
            this.push((0, sitemap_xml_1.element)(types_1.TagNames["video:family_friendly"], video.family_friendly));
          }
          if (video.restriction) {
            this.push((0, sitemap_xml_1.element)(types_1.TagNames["video:restriction"], attrBuilder(video, "restriction:relationship"), video.restriction));
          }
          if (video.gallery_loc) {
            this.push((0, sitemap_xml_1.element)(types_1.TagNames["video:gallery_loc"], { title: video["gallery_loc:title"] }, video.gallery_loc));
          }
          if (video.price) {
            this.push((0, sitemap_xml_1.element)(types_1.TagNames["video:price"], attrBuilder(video, [
              "price:resolution",
              "price:currency",
              "price:type"
            ]), video.price));
          }
          if (video.requires_subscription) {
            this.push((0, sitemap_xml_1.element)(types_1.TagNames["video:requires_subscription"], video.requires_subscription));
          }
          if (video.uploader) {
            this.push((0, sitemap_xml_1.element)(types_1.TagNames["video:uploader"], attrBuilder(video, "uploader:info"), video.uploader));
          }
          if (video.platform) {
            this.push((0, sitemap_xml_1.element)(types_1.TagNames["video:platform"], attrBuilder(video, "platform:relationship"), video.platform));
          }
          if (video.live) {
            this.push((0, sitemap_xml_1.element)(types_1.TagNames["video:live"], video.live));
          }
          if (video.id) {
            this.push((0, sitemap_xml_1.element)(types_1.TagNames["video:id"], { type: "url" }, video.id));
          }
          this.push((0, sitemap_xml_1.ctag)(types_1.TagNames["video:video"]));
        });
        item.links.forEach((link) => {
          this.push((0, sitemap_xml_1.element)(types_1.TagNames["xhtml:link"], {
            rel: "alternate",
            hreflang: link.lang || link.hreflang,
            href: link.url
          }));
        });
        if (item.expires) {
          this.push((0, sitemap_xml_1.element)(types_1.TagNames.expires, new Date(item.expires).toISOString()));
        }
        if (item.androidLink) {
          this.push((0, sitemap_xml_1.element)(types_1.TagNames["xhtml:link"], {
            rel: "alternate",
            href: item.androidLink
          }));
        }
        if (item.ampLink) {
          this.push((0, sitemap_xml_1.element)(types_1.TagNames["xhtml:link"], {
            rel: "amphtml",
            href: item.ampLink
          }));
        }
        if (item.news) {
          this.push((0, sitemap_xml_1.otag)(types_1.TagNames["news:news"]));
          this.push((0, sitemap_xml_1.otag)(types_1.TagNames["news:publication"]));
          this.push((0, sitemap_xml_1.element)(types_1.TagNames["news:name"], item.news.publication.name));
          this.push((0, sitemap_xml_1.element)(types_1.TagNames["news:language"], item.news.publication.language));
          this.push((0, sitemap_xml_1.ctag)(types_1.TagNames["news:publication"]));
          if (item.news.access) {
            this.push((0, sitemap_xml_1.element)(types_1.TagNames["news:access"], item.news.access));
          }
          if (item.news.genres) {
            this.push((0, sitemap_xml_1.element)(types_1.TagNames["news:genres"], item.news.genres));
          }
          this.push((0, sitemap_xml_1.element)(types_1.TagNames["news:publication_date"], item.news.publication_date));
          this.push((0, sitemap_xml_1.element)(types_1.TagNames["news:title"], item.news.title));
          if (item.news.keywords) {
            this.push((0, sitemap_xml_1.element)(types_1.TagNames["news:keywords"], item.news.keywords));
          }
          if (item.news.stock_tickers) {
            this.push((0, sitemap_xml_1.element)(types_1.TagNames["news:stock_tickers"], item.news.stock_tickers));
          }
          this.push((0, sitemap_xml_1.ctag)(types_1.TagNames["news:news"]));
        }
        item.img.forEach((image) => {
          this.push((0, sitemap_xml_1.otag)(types_1.TagNames["image:image"]));
          this.push((0, sitemap_xml_1.element)(types_1.TagNames["image:loc"], image.url));
          if (image.caption) {
            this.push((0, sitemap_xml_1.element)(types_1.TagNames["image:caption"], image.caption));
          }
          if (image.geoLocation) {
            this.push((0, sitemap_xml_1.element)(types_1.TagNames["image:geo_location"], image.geoLocation));
          }
          if (image.title) {
            this.push((0, sitemap_xml_1.element)(types_1.TagNames["image:title"], image.title));
          }
          if (image.license) {
            this.push((0, sitemap_xml_1.element)(types_1.TagNames["image:license"], image.license));
          }
          this.push((0, sitemap_xml_1.ctag)(types_1.TagNames["image:image"]));
        });
        this.push((0, sitemap_xml_1.ctag)(types_1.TagNames.url));
        callback();
      }
    };
    exports.SitemapItemStream = SitemapItemStream;
  }
});

// node_modules/.pnpm/sitemap@7.1.1/node_modules/sitemap/dist/lib/utils.js
var require_utils = __commonJS({
  "node_modules/.pnpm/sitemap@7.1.1/node_modules/sitemap/dist/lib/utils.js"(exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.normalizeURL = exports.chunk = exports.lineSeparatedURLsToSitemapOptions = exports.ReadlineStream = exports.mergeStreams = exports.validateSMIOptions = void 0;
    var fs_1 = require("fs");
    var stream_1 = require("stream");
    var readline_1 = require("readline");
    var url_1 = require("url");
    var types_1 = require_types();
    var errors_1 = require_errors();
    var types_2 = require_types();
    function validate(subject, name, url, level) {
      Object.keys(subject).forEach((key) => {
        const val = subject[key];
        if (types_2.validators[key] && !types_2.validators[key].test(val)) {
          if (level === types_1.ErrorLevel.THROW) {
            throw new errors_1.InvalidAttrValue(key, val, types_2.validators[key]);
          } else {
            console.warn(`${url}: ${name} key ${key} has invalid value: ${val}`);
          }
        }
      });
    }
    function handleError(error, level) {
      if (level === types_1.ErrorLevel.THROW) {
        throw error;
      } else if (level === types_1.ErrorLevel.WARN) {
        console.warn(error.name, error.message);
      }
    }
    function validateSMIOptions(conf, level = types_1.ErrorLevel.WARN, errorHandler = handleError) {
      if (!conf) {
        throw new errors_1.NoConfigError();
      }
      if (level === types_1.ErrorLevel.SILENT) {
        return conf;
      }
      const { url, changefreq, priority, news, video } = conf;
      if (!url) {
        errorHandler(new errors_1.NoURLError(), level);
      }
      if (changefreq) {
        if (!(0, types_1.isValidChangeFreq)(changefreq)) {
          errorHandler(new errors_1.ChangeFreqInvalidError(url, changefreq), level);
        }
      }
      if (priority) {
        if (!(priority >= 0 && priority <= 1)) {
          errorHandler(new errors_1.PriorityInvalidError(url, priority), level);
        }
      }
      if (news) {
        if (news.access && news.access !== "Registration" && news.access !== "Subscription") {
          errorHandler(new errors_1.InvalidNewsAccessValue(url, news.access), level);
        }
        if (!news.publication || !news.publication.name || !news.publication.language || !news.publication_date || !news.title) {
          errorHandler(new errors_1.InvalidNewsFormat(url), level);
        }
        validate(news, "news", url, level);
        validate(news.publication, "publication", url, level);
      }
      if (video) {
        video.forEach((vid) => {
          var _a;
          if (vid.duration !== void 0) {
            if (vid.duration < 0 || vid.duration > 28800) {
              errorHandler(new errors_1.InvalidVideoDuration(url, vid.duration), level);
            }
          }
          if (vid.rating !== void 0 && (vid.rating < 0 || vid.rating > 5)) {
            errorHandler(new errors_1.InvalidVideoRating(url, vid.title, vid.rating), level);
          }
          if (typeof vid !== "object" || !vid.thumbnail_loc || !vid.title || !vid.description) {
            errorHandler(new errors_1.InvalidVideoFormat(url), level);
          }
          if (vid.title.length > 100) {
            errorHandler(new errors_1.InvalidVideoTitle(url, vid.title.length), level);
          }
          if (vid.description.length > 2048) {
            errorHandler(new errors_1.InvalidVideoDescription(url, vid.description.length), level);
          }
          if (vid.view_count !== void 0 && vid.view_count < 0) {
            errorHandler(new errors_1.InvalidVideoViewCount(url, vid.view_count), level);
          }
          if (vid.tag.length > 32) {
            errorHandler(new errors_1.InvalidVideoTagCount(url, vid.tag.length), level);
          }
          if (vid.category !== void 0 && ((_a = vid.category) === null || _a === void 0 ? void 0 : _a.length) > 256) {
            errorHandler(new errors_1.InvalidVideoCategory(url, vid.category.length), level);
          }
          if (vid.family_friendly !== void 0 && !(0, types_1.isValidYesNo)(vid.family_friendly)) {
            errorHandler(new errors_1.InvalidVideoFamilyFriendly(url, vid.family_friendly), level);
          }
          if (vid.restriction) {
            if (!types_2.validators.restriction.test(vid.restriction)) {
              errorHandler(new errors_1.InvalidVideoRestriction(url, vid.restriction), level);
            }
            if (!vid["restriction:relationship"] || !(0, types_1.isAllowDeny)(vid["restriction:relationship"])) {
              errorHandler(new errors_1.InvalidVideoRestrictionRelationship(url, vid["restriction:relationship"]), level);
            }
          }
          if (vid.price === "" && vid["price:type"] === void 0 || vid["price:type"] !== void 0 && !(0, types_1.isPriceType)(vid["price:type"])) {
            errorHandler(new errors_1.InvalidVideoPriceType(url, vid["price:type"], vid.price), level);
          }
          if (vid["price:resolution"] !== void 0 && !(0, types_1.isResolution)(vid["price:resolution"])) {
            errorHandler(new errors_1.InvalidVideoResolution(url, vid["price:resolution"]), level);
          }
          if (vid["price:currency"] !== void 0 && !types_2.validators["price:currency"].test(vid["price:currency"])) {
            errorHandler(new errors_1.InvalidVideoPriceCurrency(url, vid["price:currency"]), level);
          }
          validate(vid, "video", url, level);
        });
      }
      return conf;
    }
    exports.validateSMIOptions = validateSMIOptions;
    function mergeStreams(streams, options) {
      let pass = new stream_1.PassThrough(options);
      let waiting = streams.length;
      for (const stream of streams) {
        pass = stream.pipe(pass, { end: false });
        stream.once("end", () => --waiting === 0 && pass.emit("end"));
      }
      return pass;
    }
    exports.mergeStreams = mergeStreams;
    var ReadlineStream = class extends stream_1.Readable {
      constructor(options) {
        if (options.autoDestroy === void 0) {
          options.autoDestroy = true;
        }
        options.objectMode = true;
        super(options);
        this._source = (0, readline_1.createInterface)({
          input: options.input,
          terminal: false,
          crlfDelay: Infinity
        });
        this._source.on("line", (chunk2) => {
          if (!this.push(chunk2))
            this._source.pause();
        });
        this._source.on("close", () => {
          this.push(null);
        });
      }
      _read(size) {
        this._source.resume();
      }
    };
    exports.ReadlineStream = ReadlineStream;
    function lineSeparatedURLsToSitemapOptions(stream, { isJSON } = {}) {
      return new ReadlineStream({ input: stream }).pipe(new stream_1.Transform({
        objectMode: true,
        transform: (line, encoding, cb) => {
          if (isJSON || isJSON === void 0 && line[0] === "{") {
            cb(null, JSON.parse(line));
          } else {
            cb(null, line);
          }
        }
      }));
    }
    exports.lineSeparatedURLsToSitemapOptions = lineSeparatedURLsToSitemapOptions;
    function chunk(array, size = 1) {
      size = Math.max(Math.trunc(size), 0);
      const length = array ? array.length : 0;
      if (!length || size < 1) {
        return [];
      }
      const result = Array(Math.ceil(length / size));
      let index = 0, resIndex = 0;
      while (index < length) {
        result[resIndex++] = array.slice(index, index += size);
      }
      return result;
    }
    exports.chunk = chunk;
    function boolToYESNO(bool) {
      if (bool === void 0) {
        return bool;
      }
      if (typeof bool === "boolean") {
        return bool ? types_1.EnumYesNo.yes : types_1.EnumYesNo.no;
      }
      return bool;
    }
    function normalizeURL(elem, hostname, lastmodDateOnly = false) {
      let smi = {
        img: [],
        video: [],
        links: [],
        url: ""
      };
      let smiLoose;
      if (typeof elem === "string") {
        smi.url = elem;
        smiLoose = { url: elem };
      } else {
        smiLoose = elem;
      }
      smi.url = new url_1.URL(smiLoose.url, hostname).toString();
      let img = [];
      if (smiLoose.img) {
        if (typeof smiLoose.img === "string") {
          smiLoose.img = [{ url: smiLoose.img }];
        } else if (!Array.isArray(smiLoose.img)) {
          smiLoose.img = [smiLoose.img];
        }
        img = smiLoose.img.map((el) => typeof el === "string" ? { url: el } : el);
      }
      smi.img = img.map((el) => ({
        ...el,
        url: new url_1.URL(el.url, hostname).toString()
      }));
      let links = [];
      if (smiLoose.links) {
        links = smiLoose.links;
      }
      smi.links = links.map((link) => {
        return { ...link, url: new url_1.URL(link.url, hostname).toString() };
      });
      if (smiLoose.video) {
        if (!Array.isArray(smiLoose.video)) {
          smiLoose.video = [smiLoose.video];
        }
        smi.video = smiLoose.video.map((video) => {
          const nv = {
            ...video,
            family_friendly: boolToYESNO(video.family_friendly),
            live: boolToYESNO(video.live),
            requires_subscription: boolToYESNO(video.requires_subscription),
            tag: [],
            rating: void 0
          };
          if (video.tag !== void 0) {
            nv.tag = !Array.isArray(video.tag) ? [video.tag] : video.tag;
          }
          if (video.rating !== void 0) {
            if (typeof video.rating === "string") {
              nv.rating = parseFloat(video.rating);
            } else {
              nv.rating = video.rating;
            }
          }
          if (typeof video.view_count === "string") {
            nv.view_count = parseInt(video.view_count, 10);
          } else if (typeof video.view_count === "number") {
            nv.view_count = video.view_count;
          }
          return nv;
        });
      }
      if (smiLoose.lastmodfile) {
        const { mtime } = (0, fs_1.statSync)(smiLoose.lastmodfile);
        smi.lastmod = new Date(mtime).toISOString();
      } else if (smiLoose.lastmodISO) {
        smi.lastmod = new Date(smiLoose.lastmodISO).toISOString();
      } else if (smiLoose.lastmod) {
        smi.lastmod = new Date(smiLoose.lastmod).toISOString();
      }
      if (lastmodDateOnly && smi.lastmod) {
        smi.lastmod = smi.lastmod.slice(0, 10);
      }
      delete smiLoose.lastmodfile;
      delete smiLoose.lastmodISO;
      smi = { ...smiLoose, ...smi };
      return smi;
    }
    exports.normalizeURL = normalizeURL;
  }
});

// node_modules/.pnpm/sitemap@7.1.1/node_modules/sitemap/dist/lib/sitemap-stream.js
var require_sitemap_stream = __commonJS({
  "node_modules/.pnpm/sitemap@7.1.1/node_modules/sitemap/dist/lib/sitemap-stream.js"(exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.streamToPromise = exports.SitemapStream = exports.closetag = exports.stylesheetInclude = void 0;
    var stream_1 = require("stream");
    var types_1 = require_types();
    var utils_1 = require_utils();
    var sitemap_item_stream_1 = require_sitemap_item_stream();
    var errors_1 = require_errors();
    var xmlDec = '<?xml version="1.0" encoding="UTF-8"?>';
    var stylesheetInclude = (url) => {
      return `<?xml-stylesheet type="text/xsl" href="${url}"?>`;
    };
    exports.stylesheetInclude = stylesheetInclude;
    var urlsetTagStart = '<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9"';
    var getURLSetNs = ({ news, video, image, xhtml, custom }, xslURL) => {
      let ns = xmlDec;
      if (xslURL) {
        ns += (0, exports.stylesheetInclude)(xslURL);
      }
      ns += urlsetTagStart;
      if (news) {
        ns += ' xmlns:news="http://www.google.com/schemas/sitemap-news/0.9"';
      }
      if (xhtml) {
        ns += ' xmlns:xhtml="http://www.w3.org/1999/xhtml"';
      }
      if (image) {
        ns += ' xmlns:image="http://www.google.com/schemas/sitemap-image/1.1"';
      }
      if (video) {
        ns += ' xmlns:video="http://www.google.com/schemas/sitemap-video/1.1"';
      }
      if (custom) {
        ns += " " + custom.join(" ");
      }
      return ns + ">";
    };
    exports.closetag = "</urlset>";
    var defaultXMLNS = {
      news: true,
      xhtml: true,
      image: true,
      video: true
    };
    var defaultStreamOpts = {
      xmlns: defaultXMLNS
    };
    var SitemapStream2 = class extends stream_1.Transform {
      constructor(opts = defaultStreamOpts) {
        opts.objectMode = true;
        super(opts);
        this.hasHeadOutput = false;
        this.hostname = opts.hostname;
        this.level = opts.level || types_1.ErrorLevel.WARN;
        this.errorHandler = opts.errorHandler;
        this.smiStream = new sitemap_item_stream_1.SitemapItemStream({ level: opts.level });
        this.smiStream.on("data", (data) => this.push(data));
        this.lastmodDateOnly = opts.lastmodDateOnly || false;
        this.xmlNS = opts.xmlns || defaultXMLNS;
        this.xslUrl = opts.xslUrl;
      }
      _transform(item, encoding, callback) {
        if (!this.hasHeadOutput) {
          this.hasHeadOutput = true;
          this.push(getURLSetNs(this.xmlNS, this.xslUrl));
        }
        this.smiStream.write((0, utils_1.validateSMIOptions)((0, utils_1.normalizeURL)(item, this.hostname, this.lastmodDateOnly), this.level, this.errorHandler));
        callback();
      }
      _flush(cb) {
        if (!this.hasHeadOutput) {
          cb(new errors_1.EmptySitemap());
        } else {
          this.push(exports.closetag);
          cb();
        }
      }
    };
    exports.SitemapStream = SitemapStream2;
    function streamToPromise2(stream) {
      return new Promise((resolve2, reject) => {
        const drain = [];
        stream.pipe(new stream_1.Writable({
          write(chunk, enc, next) {
            drain.push(chunk);
            next();
          }
        })).on("error", reject).on("finish", () => {
          if (!drain.length) {
            reject(new errors_1.EmptyStream());
          } else {
            resolve2(Buffer.concat(drain));
          }
        });
      });
    }
    exports.streamToPromise = streamToPromise2;
  }
});

// node_modules/.pnpm/sitemap@7.1.1/node_modules/sitemap/dist/lib/sitemap-index-stream.js
var require_sitemap_index_stream = __commonJS({
  "node_modules/.pnpm/sitemap@7.1.1/node_modules/sitemap/dist/lib/sitemap-index-stream.js"(exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.SitemapAndIndexStream = exports.SitemapIndexStream = exports.IndexTagNames = void 0;
    var stream_1 = require("stream");
    var types_1 = require_types();
    var sitemap_stream_1 = require_sitemap_stream();
    var sitemap_xml_1 = require_sitemap_xml();
    var IndexTagNames;
    (function(IndexTagNames2) {
      IndexTagNames2["sitemap"] = "sitemap";
      IndexTagNames2["loc"] = "loc";
      IndexTagNames2["lastmod"] = "lastmod";
    })(IndexTagNames = exports.IndexTagNames || (exports.IndexTagNames = {}));
    var xmlDec = '<?xml version="1.0" encoding="UTF-8"?>';
    var sitemapIndexTagStart = '<sitemapindex xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">';
    var closetag = "</sitemapindex>";
    var defaultStreamOpts = {};
    var SitemapIndexStream = class extends stream_1.Transform {
      constructor(opts = defaultStreamOpts) {
        var _a;
        opts.objectMode = true;
        super(opts);
        this.hasHeadOutput = false;
        this.lastmodDateOnly = opts.lastmodDateOnly || false;
        this.level = (_a = opts.level) !== null && _a !== void 0 ? _a : types_1.ErrorLevel.WARN;
        this.xslUrl = opts.xslUrl;
      }
      _transform(item, encoding, callback) {
        if (!this.hasHeadOutput) {
          this.hasHeadOutput = true;
          let stylesheet = "";
          if (this.xslUrl) {
            stylesheet = (0, sitemap_stream_1.stylesheetInclude)(this.xslUrl);
          }
          this.push(xmlDec + stylesheet + sitemapIndexTagStart);
        }
        this.push((0, sitemap_xml_1.otag)(IndexTagNames.sitemap));
        if (typeof item === "string") {
          this.push((0, sitemap_xml_1.element)(IndexTagNames.loc, item));
        } else {
          this.push((0, sitemap_xml_1.element)(IndexTagNames.loc, item.url));
          if (item.lastmod) {
            const lastmod = new Date(item.lastmod).toISOString();
            this.push((0, sitemap_xml_1.element)(IndexTagNames.lastmod, this.lastmodDateOnly ? lastmod.slice(0, 10) : lastmod));
          }
        }
        this.push((0, sitemap_xml_1.ctag)(IndexTagNames.sitemap));
        callback();
      }
      _flush(cb) {
        this.push(closetag);
        cb();
      }
    };
    exports.SitemapIndexStream = SitemapIndexStream;
    var SitemapAndIndexStream = class extends SitemapIndexStream {
      constructor(opts) {
        var _a;
        opts.objectMode = true;
        super(opts);
        this.i = 0;
        this.getSitemapStream = opts.getSitemapStream;
        [this.idxItem, this.currentSitemap, this.currentSitemapPipeline] = this.getSitemapStream(0);
        this.limit = (_a = opts.limit) !== null && _a !== void 0 ? _a : 45e3;
      }
      _writeSMI(item) {
        this.currentSitemap.write(item);
        this.i++;
      }
      _transform(item, encoding, callback) {
        var _a;
        if (this.i === 0) {
          this._writeSMI(item);
          super._transform(this.idxItem, encoding, callback);
        } else if (this.i % this.limit === 0) {
          const onFinish = () => {
            const [idxItem, currentSitemap, currentSitemapPipeline] = this.getSitemapStream(this.i / this.limit);
            this.currentSitemap = currentSitemap;
            this.currentSitemapPipeline = currentSitemapPipeline;
            this._writeSMI(item);
            super._transform(idxItem, encoding, callback);
          };
          (_a = this.currentSitemapPipeline) === null || _a === void 0 ? void 0 : _a.on("finish", onFinish);
          this.currentSitemap.end(!this.currentSitemapPipeline ? onFinish : void 0);
        } else {
          this._writeSMI(item);
          callback();
        }
      }
      _flush(cb) {
        var _a;
        const onFinish = () => super._flush(cb);
        (_a = this.currentSitemapPipeline) === null || _a === void 0 ? void 0 : _a.on("finish", onFinish);
        this.currentSitemap.end(!this.currentSitemapPipeline ? onFinish : void 0);
      }
    };
    exports.SitemapAndIndexStream = SitemapAndIndexStream;
  }
});

// node_modules/.pnpm/sitemap@7.1.1/node_modules/sitemap/dist/lib/xmllint.js
var require_xmllint = __commonJS({
  "node_modules/.pnpm/sitemap@7.1.1/node_modules/sitemap/dist/lib/xmllint.js"(exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.xmlLint = void 0;
    var path_1 = require("path");
    var child_process_1 = require("child_process");
    var errors_1 = require_errors();
    function xmlLint(xml) {
      const args = [
        "--schema",
        (0, path_1.resolve)(__dirname, "..", "..", "schema", "all.xsd"),
        "--noout",
        "-"
      ];
      if (typeof xml === "string") {
        args[args.length - 1] = xml;
      }
      return new Promise((resolve2, reject) => {
        (0, child_process_1.execFile)("which", ["xmllint"], (error, stdout, stderr) => {
          if (error) {
            reject([new errors_1.XMLLintUnavailable()]);
            return;
          }
          const xmllint = (0, child_process_1.execFile)("xmllint", args, (error2, stdout2, stderr2) => {
            if (error2) {
              reject([error2, stderr2]);
            }
            resolve2();
          });
          if (xmllint.stdout) {
            xmllint.stdout.unpipe();
            if (typeof xml !== "string" && xml && xmllint.stdin) {
              xml.pipe(xmllint.stdin);
            }
          }
        });
      });
    }
    exports.xmlLint = xmlLint;
  }
});

// node_modules/.pnpm/sax@1.2.4/node_modules/sax/lib/sax.js
var require_sax = __commonJS({
  "node_modules/.pnpm/sax@1.2.4/node_modules/sax/lib/sax.js"(exports) {
    (function(sax) {
      sax.parser = function(strict, opt) {
        return new SAXParser(strict, opt);
      };
      sax.SAXParser = SAXParser;
      sax.SAXStream = SAXStream;
      sax.createStream = createStream;
      sax.MAX_BUFFER_LENGTH = 64 * 1024;
      var buffers = [
        "comment",
        "sgmlDecl",
        "textNode",
        "tagName",
        "doctype",
        "procInstName",
        "procInstBody",
        "entity",
        "attribName",
        "attribValue",
        "cdata",
        "script"
      ];
      sax.EVENTS = [
        "text",
        "processinginstruction",
        "sgmldeclaration",
        "doctype",
        "comment",
        "opentagstart",
        "attribute",
        "opentag",
        "closetag",
        "opencdata",
        "cdata",
        "closecdata",
        "error",
        "end",
        "ready",
        "script",
        "opennamespace",
        "closenamespace"
      ];
      function SAXParser(strict, opt) {
        if (!(this instanceof SAXParser)) {
          return new SAXParser(strict, opt);
        }
        var parser = this;
        clearBuffers(parser);
        parser.q = parser.c = "";
        parser.bufferCheckPosition = sax.MAX_BUFFER_LENGTH;
        parser.opt = opt || {};
        parser.opt.lowercase = parser.opt.lowercase || parser.opt.lowercasetags;
        parser.looseCase = parser.opt.lowercase ? "toLowerCase" : "toUpperCase";
        parser.tags = [];
        parser.closed = parser.closedRoot = parser.sawRoot = false;
        parser.tag = parser.error = null;
        parser.strict = !!strict;
        parser.noscript = !!(strict || parser.opt.noscript);
        parser.state = S.BEGIN;
        parser.strictEntities = parser.opt.strictEntities;
        parser.ENTITIES = parser.strictEntities ? Object.create(sax.XML_ENTITIES) : Object.create(sax.ENTITIES);
        parser.attribList = [];
        if (parser.opt.xmlns) {
          parser.ns = Object.create(rootNS);
        }
        parser.trackPosition = parser.opt.position !== false;
        if (parser.trackPosition) {
          parser.position = parser.line = parser.column = 0;
        }
        emit(parser, "onready");
      }
      if (!Object.create) {
        Object.create = function(o) {
          function F() {
          }
          F.prototype = o;
          var newf = new F();
          return newf;
        };
      }
      if (!Object.keys) {
        Object.keys = function(o) {
          var a = [];
          for (var i in o)
            if (o.hasOwnProperty(i))
              a.push(i);
          return a;
        };
      }
      function checkBufferLength(parser) {
        var maxAllowed = Math.max(sax.MAX_BUFFER_LENGTH, 10);
        var maxActual = 0;
        for (var i = 0, l = buffers.length; i < l; i++) {
          var len = parser[buffers[i]].length;
          if (len > maxAllowed) {
            switch (buffers[i]) {
              case "textNode":
                closeText(parser);
                break;
              case "cdata":
                emitNode(parser, "oncdata", parser.cdata);
                parser.cdata = "";
                break;
              case "script":
                emitNode(parser, "onscript", parser.script);
                parser.script = "";
                break;
              default:
                error(parser, "Max buffer length exceeded: " + buffers[i]);
            }
          }
          maxActual = Math.max(maxActual, len);
        }
        var m = sax.MAX_BUFFER_LENGTH - maxActual;
        parser.bufferCheckPosition = m + parser.position;
      }
      function clearBuffers(parser) {
        for (var i = 0, l = buffers.length; i < l; i++) {
          parser[buffers[i]] = "";
        }
      }
      function flushBuffers(parser) {
        closeText(parser);
        if (parser.cdata !== "") {
          emitNode(parser, "oncdata", parser.cdata);
          parser.cdata = "";
        }
        if (parser.script !== "") {
          emitNode(parser, "onscript", parser.script);
          parser.script = "";
        }
      }
      SAXParser.prototype = {
        end: function() {
          end(this);
        },
        write,
        resume: function() {
          this.error = null;
          return this;
        },
        close: function() {
          return this.write(null);
        },
        flush: function() {
          flushBuffers(this);
        }
      };
      var Stream;
      try {
        Stream = require("stream").Stream;
      } catch (ex) {
        Stream = function() {
        };
      }
      var streamWraps = sax.EVENTS.filter(function(ev) {
        return ev !== "error" && ev !== "end";
      });
      function createStream(strict, opt) {
        return new SAXStream(strict, opt);
      }
      function SAXStream(strict, opt) {
        if (!(this instanceof SAXStream)) {
          return new SAXStream(strict, opt);
        }
        Stream.apply(this);
        this._parser = new SAXParser(strict, opt);
        this.writable = true;
        this.readable = true;
        var me = this;
        this._parser.onend = function() {
          me.emit("end");
        };
        this._parser.onerror = function(er) {
          me.emit("error", er);
          me._parser.error = null;
        };
        this._decoder = null;
        streamWraps.forEach(function(ev) {
          Object.defineProperty(me, "on" + ev, {
            get: function() {
              return me._parser["on" + ev];
            },
            set: function(h) {
              if (!h) {
                me.removeAllListeners(ev);
                me._parser["on" + ev] = h;
                return h;
              }
              me.on(ev, h);
            },
            enumerable: true,
            configurable: false
          });
        });
      }
      SAXStream.prototype = Object.create(Stream.prototype, {
        constructor: {
          value: SAXStream
        }
      });
      SAXStream.prototype.write = function(data) {
        if (typeof Buffer === "function" && typeof Buffer.isBuffer === "function" && Buffer.isBuffer(data)) {
          if (!this._decoder) {
            var SD = require("string_decoder").StringDecoder;
            this._decoder = new SD("utf8");
          }
          data = this._decoder.write(data);
        }
        this._parser.write(data.toString());
        this.emit("data", data);
        return true;
      };
      SAXStream.prototype.end = function(chunk) {
        if (chunk && chunk.length) {
          this.write(chunk);
        }
        this._parser.end();
        return true;
      };
      SAXStream.prototype.on = function(ev, handler) {
        var me = this;
        if (!me._parser["on" + ev] && streamWraps.indexOf(ev) !== -1) {
          me._parser["on" + ev] = function() {
            var args = arguments.length === 1 ? [arguments[0]] : Array.apply(null, arguments);
            args.splice(0, 0, ev);
            me.emit.apply(me, args);
          };
        }
        return Stream.prototype.on.call(me, ev, handler);
      };
      var CDATA = "[CDATA[";
      var DOCTYPE = "DOCTYPE";
      var XML_NAMESPACE = "http://www.w3.org/XML/1998/namespace";
      var XMLNS_NAMESPACE = "http://www.w3.org/2000/xmlns/";
      var rootNS = { xml: XML_NAMESPACE, xmlns: XMLNS_NAMESPACE };
      var nameStart = /[:_A-Za-z\u00C0-\u00D6\u00D8-\u00F6\u00F8-\u02FF\u0370-\u037D\u037F-\u1FFF\u200C-\u200D\u2070-\u218F\u2C00-\u2FEF\u3001-\uD7FF\uF900-\uFDCF\uFDF0-\uFFFD]/;
      var nameBody = /[:_A-Za-z\u00C0-\u00D6\u00D8-\u00F6\u00F8-\u02FF\u0370-\u037D\u037F-\u1FFF\u200C-\u200D\u2070-\u218F\u2C00-\u2FEF\u3001-\uD7FF\uF900-\uFDCF\uFDF0-\uFFFD\u00B7\u0300-\u036F\u203F-\u2040.\d-]/;
      var entityStart = /[#:_A-Za-z\u00C0-\u00D6\u00D8-\u00F6\u00F8-\u02FF\u0370-\u037D\u037F-\u1FFF\u200C-\u200D\u2070-\u218F\u2C00-\u2FEF\u3001-\uD7FF\uF900-\uFDCF\uFDF0-\uFFFD]/;
      var entityBody = /[#:_A-Za-z\u00C0-\u00D6\u00D8-\u00F6\u00F8-\u02FF\u0370-\u037D\u037F-\u1FFF\u200C-\u200D\u2070-\u218F\u2C00-\u2FEF\u3001-\uD7FF\uF900-\uFDCF\uFDF0-\uFFFD\u00B7\u0300-\u036F\u203F-\u2040.\d-]/;
      function isWhitespace(c) {
        return c === " " || c === "\n" || c === "\r" || c === "	";
      }
      function isQuote(c) {
        return c === '"' || c === "'";
      }
      function isAttribEnd(c) {
        return c === ">" || isWhitespace(c);
      }
      function isMatch(regex, c) {
        return regex.test(c);
      }
      function notMatch(regex, c) {
        return !isMatch(regex, c);
      }
      var S = 0;
      sax.STATE = {
        BEGIN: S++,
        BEGIN_WHITESPACE: S++,
        TEXT: S++,
        TEXT_ENTITY: S++,
        OPEN_WAKA: S++,
        SGML_DECL: S++,
        SGML_DECL_QUOTED: S++,
        DOCTYPE: S++,
        DOCTYPE_QUOTED: S++,
        DOCTYPE_DTD: S++,
        DOCTYPE_DTD_QUOTED: S++,
        COMMENT_STARTING: S++,
        COMMENT: S++,
        COMMENT_ENDING: S++,
        COMMENT_ENDED: S++,
        CDATA: S++,
        CDATA_ENDING: S++,
        CDATA_ENDING_2: S++,
        PROC_INST: S++,
        PROC_INST_BODY: S++,
        PROC_INST_ENDING: S++,
        OPEN_TAG: S++,
        OPEN_TAG_SLASH: S++,
        ATTRIB: S++,
        ATTRIB_NAME: S++,
        ATTRIB_NAME_SAW_WHITE: S++,
        ATTRIB_VALUE: S++,
        ATTRIB_VALUE_QUOTED: S++,
        ATTRIB_VALUE_CLOSED: S++,
        ATTRIB_VALUE_UNQUOTED: S++,
        ATTRIB_VALUE_ENTITY_Q: S++,
        ATTRIB_VALUE_ENTITY_U: S++,
        CLOSE_TAG: S++,
        CLOSE_TAG_SAW_WHITE: S++,
        SCRIPT: S++,
        SCRIPT_ENDING: S++
      };
      sax.XML_ENTITIES = {
        "amp": "&",
        "gt": ">",
        "lt": "<",
        "quot": '"',
        "apos": "'"
      };
      sax.ENTITIES = {
        "amp": "&",
        "gt": ">",
        "lt": "<",
        "quot": '"',
        "apos": "'",
        "AElig": 198,
        "Aacute": 193,
        "Acirc": 194,
        "Agrave": 192,
        "Aring": 197,
        "Atilde": 195,
        "Auml": 196,
        "Ccedil": 199,
        "ETH": 208,
        "Eacute": 201,
        "Ecirc": 202,
        "Egrave": 200,
        "Euml": 203,
        "Iacute": 205,
        "Icirc": 206,
        "Igrave": 204,
        "Iuml": 207,
        "Ntilde": 209,
        "Oacute": 211,
        "Ocirc": 212,
        "Ograve": 210,
        "Oslash": 216,
        "Otilde": 213,
        "Ouml": 214,
        "THORN": 222,
        "Uacute": 218,
        "Ucirc": 219,
        "Ugrave": 217,
        "Uuml": 220,
        "Yacute": 221,
        "aacute": 225,
        "acirc": 226,
        "aelig": 230,
        "agrave": 224,
        "aring": 229,
        "atilde": 227,
        "auml": 228,
        "ccedil": 231,
        "eacute": 233,
        "ecirc": 234,
        "egrave": 232,
        "eth": 240,
        "euml": 235,
        "iacute": 237,
        "icirc": 238,
        "igrave": 236,
        "iuml": 239,
        "ntilde": 241,
        "oacute": 243,
        "ocirc": 244,
        "ograve": 242,
        "oslash": 248,
        "otilde": 245,
        "ouml": 246,
        "szlig": 223,
        "thorn": 254,
        "uacute": 250,
        "ucirc": 251,
        "ugrave": 249,
        "uuml": 252,
        "yacute": 253,
        "yuml": 255,
        "copy": 169,
        "reg": 174,
        "nbsp": 160,
        "iexcl": 161,
        "cent": 162,
        "pound": 163,
        "curren": 164,
        "yen": 165,
        "brvbar": 166,
        "sect": 167,
        "uml": 168,
        "ordf": 170,
        "laquo": 171,
        "not": 172,
        "shy": 173,
        "macr": 175,
        "deg": 176,
        "plusmn": 177,
        "sup1": 185,
        "sup2": 178,
        "sup3": 179,
        "acute": 180,
        "micro": 181,
        "para": 182,
        "middot": 183,
        "cedil": 184,
        "ordm": 186,
        "raquo": 187,
        "frac14": 188,
        "frac12": 189,
        "frac34": 190,
        "iquest": 191,
        "times": 215,
        "divide": 247,
        "OElig": 338,
        "oelig": 339,
        "Scaron": 352,
        "scaron": 353,
        "Yuml": 376,
        "fnof": 402,
        "circ": 710,
        "tilde": 732,
        "Alpha": 913,
        "Beta": 914,
        "Gamma": 915,
        "Delta": 916,
        "Epsilon": 917,
        "Zeta": 918,
        "Eta": 919,
        "Theta": 920,
        "Iota": 921,
        "Kappa": 922,
        "Lambda": 923,
        "Mu": 924,
        "Nu": 925,
        "Xi": 926,
        "Omicron": 927,
        "Pi": 928,
        "Rho": 929,
        "Sigma": 931,
        "Tau": 932,
        "Upsilon": 933,
        "Phi": 934,
        "Chi": 935,
        "Psi": 936,
        "Omega": 937,
        "alpha": 945,
        "beta": 946,
        "gamma": 947,
        "delta": 948,
        "epsilon": 949,
        "zeta": 950,
        "eta": 951,
        "theta": 952,
        "iota": 953,
        "kappa": 954,
        "lambda": 955,
        "mu": 956,
        "nu": 957,
        "xi": 958,
        "omicron": 959,
        "pi": 960,
        "rho": 961,
        "sigmaf": 962,
        "sigma": 963,
        "tau": 964,
        "upsilon": 965,
        "phi": 966,
        "chi": 967,
        "psi": 968,
        "omega": 969,
        "thetasym": 977,
        "upsih": 978,
        "piv": 982,
        "ensp": 8194,
        "emsp": 8195,
        "thinsp": 8201,
        "zwnj": 8204,
        "zwj": 8205,
        "lrm": 8206,
        "rlm": 8207,
        "ndash": 8211,
        "mdash": 8212,
        "lsquo": 8216,
        "rsquo": 8217,
        "sbquo": 8218,
        "ldquo": 8220,
        "rdquo": 8221,
        "bdquo": 8222,
        "dagger": 8224,
        "Dagger": 8225,
        "bull": 8226,
        "hellip": 8230,
        "permil": 8240,
        "prime": 8242,
        "Prime": 8243,
        "lsaquo": 8249,
        "rsaquo": 8250,
        "oline": 8254,
        "frasl": 8260,
        "euro": 8364,
        "image": 8465,
        "weierp": 8472,
        "real": 8476,
        "trade": 8482,
        "alefsym": 8501,
        "larr": 8592,
        "uarr": 8593,
        "rarr": 8594,
        "darr": 8595,
        "harr": 8596,
        "crarr": 8629,
        "lArr": 8656,
        "uArr": 8657,
        "rArr": 8658,
        "dArr": 8659,
        "hArr": 8660,
        "forall": 8704,
        "part": 8706,
        "exist": 8707,
        "empty": 8709,
        "nabla": 8711,
        "isin": 8712,
        "notin": 8713,
        "ni": 8715,
        "prod": 8719,
        "sum": 8721,
        "minus": 8722,
        "lowast": 8727,
        "radic": 8730,
        "prop": 8733,
        "infin": 8734,
        "ang": 8736,
        "and": 8743,
        "or": 8744,
        "cap": 8745,
        "cup": 8746,
        "int": 8747,
        "there4": 8756,
        "sim": 8764,
        "cong": 8773,
        "asymp": 8776,
        "ne": 8800,
        "equiv": 8801,
        "le": 8804,
        "ge": 8805,
        "sub": 8834,
        "sup": 8835,
        "nsub": 8836,
        "sube": 8838,
        "supe": 8839,
        "oplus": 8853,
        "otimes": 8855,
        "perp": 8869,
        "sdot": 8901,
        "lceil": 8968,
        "rceil": 8969,
        "lfloor": 8970,
        "rfloor": 8971,
        "lang": 9001,
        "rang": 9002,
        "loz": 9674,
        "spades": 9824,
        "clubs": 9827,
        "hearts": 9829,
        "diams": 9830
      };
      Object.keys(sax.ENTITIES).forEach(function(key) {
        var e = sax.ENTITIES[key];
        var s2 = typeof e === "number" ? String.fromCharCode(e) : e;
        sax.ENTITIES[key] = s2;
      });
      for (var s in sax.STATE) {
        sax.STATE[sax.STATE[s]] = s;
      }
      S = sax.STATE;
      function emit(parser, event, data) {
        parser[event] && parser[event](data);
      }
      function emitNode(parser, nodeType, data) {
        if (parser.textNode)
          closeText(parser);
        emit(parser, nodeType, data);
      }
      function closeText(parser) {
        parser.textNode = textopts(parser.opt, parser.textNode);
        if (parser.textNode)
          emit(parser, "ontext", parser.textNode);
        parser.textNode = "";
      }
      function textopts(opt, text) {
        if (opt.trim)
          text = text.trim();
        if (opt.normalize)
          text = text.replace(/\s+/g, " ");
        return text;
      }
      function error(parser, er) {
        closeText(parser);
        if (parser.trackPosition) {
          er += "\nLine: " + parser.line + "\nColumn: " + parser.column + "\nChar: " + parser.c;
        }
        er = new Error(er);
        parser.error = er;
        emit(parser, "onerror", er);
        return parser;
      }
      function end(parser) {
        if (parser.sawRoot && !parser.closedRoot)
          strictFail(parser, "Unclosed root tag");
        if (parser.state !== S.BEGIN && parser.state !== S.BEGIN_WHITESPACE && parser.state !== S.TEXT) {
          error(parser, "Unexpected end");
        }
        closeText(parser);
        parser.c = "";
        parser.closed = true;
        emit(parser, "onend");
        SAXParser.call(parser, parser.strict, parser.opt);
        return parser;
      }
      function strictFail(parser, message) {
        if (typeof parser !== "object" || !(parser instanceof SAXParser)) {
          throw new Error("bad call to strictFail");
        }
        if (parser.strict) {
          error(parser, message);
        }
      }
      function newTag(parser) {
        if (!parser.strict)
          parser.tagName = parser.tagName[parser.looseCase]();
        var parent = parser.tags[parser.tags.length - 1] || parser;
        var tag = parser.tag = { name: parser.tagName, attributes: {} };
        if (parser.opt.xmlns) {
          tag.ns = parent.ns;
        }
        parser.attribList.length = 0;
        emitNode(parser, "onopentagstart", tag);
      }
      function qname(name, attribute) {
        var i = name.indexOf(":");
        var qualName = i < 0 ? ["", name] : name.split(":");
        var prefix = qualName[0];
        var local = qualName[1];
        if (attribute && name === "xmlns") {
          prefix = "xmlns";
          local = "";
        }
        return { prefix, local };
      }
      function attrib(parser) {
        if (!parser.strict) {
          parser.attribName = parser.attribName[parser.looseCase]();
        }
        if (parser.attribList.indexOf(parser.attribName) !== -1 || parser.tag.attributes.hasOwnProperty(parser.attribName)) {
          parser.attribName = parser.attribValue = "";
          return;
        }
        if (parser.opt.xmlns) {
          var qn = qname(parser.attribName, true);
          var prefix = qn.prefix;
          var local = qn.local;
          if (prefix === "xmlns") {
            if (local === "xml" && parser.attribValue !== XML_NAMESPACE) {
              strictFail(
                parser,
                "xml: prefix must be bound to " + XML_NAMESPACE + "\nActual: " + parser.attribValue
              );
            } else if (local === "xmlns" && parser.attribValue !== XMLNS_NAMESPACE) {
              strictFail(
                parser,
                "xmlns: prefix must be bound to " + XMLNS_NAMESPACE + "\nActual: " + parser.attribValue
              );
            } else {
              var tag = parser.tag;
              var parent = parser.tags[parser.tags.length - 1] || parser;
              if (tag.ns === parent.ns) {
                tag.ns = Object.create(parent.ns);
              }
              tag.ns[local] = parser.attribValue;
            }
          }
          parser.attribList.push([parser.attribName, parser.attribValue]);
        } else {
          parser.tag.attributes[parser.attribName] = parser.attribValue;
          emitNode(parser, "onattribute", {
            name: parser.attribName,
            value: parser.attribValue
          });
        }
        parser.attribName = parser.attribValue = "";
      }
      function openTag(parser, selfClosing) {
        if (parser.opt.xmlns) {
          var tag = parser.tag;
          var qn = qname(parser.tagName);
          tag.prefix = qn.prefix;
          tag.local = qn.local;
          tag.uri = tag.ns[qn.prefix] || "";
          if (tag.prefix && !tag.uri) {
            strictFail(parser, "Unbound namespace prefix: " + JSON.stringify(parser.tagName));
            tag.uri = qn.prefix;
          }
          var parent = parser.tags[parser.tags.length - 1] || parser;
          if (tag.ns && parent.ns !== tag.ns) {
            Object.keys(tag.ns).forEach(function(p) {
              emitNode(parser, "onopennamespace", {
                prefix: p,
                uri: tag.ns[p]
              });
            });
          }
          for (var i = 0, l = parser.attribList.length; i < l; i++) {
            var nv = parser.attribList[i];
            var name = nv[0];
            var value = nv[1];
            var qualName = qname(name, true);
            var prefix = qualName.prefix;
            var local = qualName.local;
            var uri = prefix === "" ? "" : tag.ns[prefix] || "";
            var a = {
              name,
              value,
              prefix,
              local,
              uri
            };
            if (prefix && prefix !== "xmlns" && !uri) {
              strictFail(parser, "Unbound namespace prefix: " + JSON.stringify(prefix));
              a.uri = prefix;
            }
            parser.tag.attributes[name] = a;
            emitNode(parser, "onattribute", a);
          }
          parser.attribList.length = 0;
        }
        parser.tag.isSelfClosing = !!selfClosing;
        parser.sawRoot = true;
        parser.tags.push(parser.tag);
        emitNode(parser, "onopentag", parser.tag);
        if (!selfClosing) {
          if (!parser.noscript && parser.tagName.toLowerCase() === "script") {
            parser.state = S.SCRIPT;
          } else {
            parser.state = S.TEXT;
          }
          parser.tag = null;
          parser.tagName = "";
        }
        parser.attribName = parser.attribValue = "";
        parser.attribList.length = 0;
      }
      function closeTag(parser) {
        if (!parser.tagName) {
          strictFail(parser, "Weird empty close tag.");
          parser.textNode += "</>";
          parser.state = S.TEXT;
          return;
        }
        if (parser.script) {
          if (parser.tagName !== "script") {
            parser.script += "</" + parser.tagName + ">";
            parser.tagName = "";
            parser.state = S.SCRIPT;
            return;
          }
          emitNode(parser, "onscript", parser.script);
          parser.script = "";
        }
        var t = parser.tags.length;
        var tagName = parser.tagName;
        if (!parser.strict) {
          tagName = tagName[parser.looseCase]();
        }
        var closeTo = tagName;
        while (t--) {
          var close = parser.tags[t];
          if (close.name !== closeTo) {
            strictFail(parser, "Unexpected close tag");
          } else {
            break;
          }
        }
        if (t < 0) {
          strictFail(parser, "Unmatched closing tag: " + parser.tagName);
          parser.textNode += "</" + parser.tagName + ">";
          parser.state = S.TEXT;
          return;
        }
        parser.tagName = tagName;
        var s2 = parser.tags.length;
        while (s2-- > t) {
          var tag = parser.tag = parser.tags.pop();
          parser.tagName = parser.tag.name;
          emitNode(parser, "onclosetag", parser.tagName);
          var x = {};
          for (var i in tag.ns) {
            x[i] = tag.ns[i];
          }
          var parent = parser.tags[parser.tags.length - 1] || parser;
          if (parser.opt.xmlns && tag.ns !== parent.ns) {
            Object.keys(tag.ns).forEach(function(p) {
              var n = tag.ns[p];
              emitNode(parser, "onclosenamespace", { prefix: p, uri: n });
            });
          }
        }
        if (t === 0)
          parser.closedRoot = true;
        parser.tagName = parser.attribValue = parser.attribName = "";
        parser.attribList.length = 0;
        parser.state = S.TEXT;
      }
      function parseEntity(parser) {
        var entity = parser.entity;
        var entityLC = entity.toLowerCase();
        var num;
        var numStr = "";
        if (parser.ENTITIES[entity]) {
          return parser.ENTITIES[entity];
        }
        if (parser.ENTITIES[entityLC]) {
          return parser.ENTITIES[entityLC];
        }
        entity = entityLC;
        if (entity.charAt(0) === "#") {
          if (entity.charAt(1) === "x") {
            entity = entity.slice(2);
            num = parseInt(entity, 16);
            numStr = num.toString(16);
          } else {
            entity = entity.slice(1);
            num = parseInt(entity, 10);
            numStr = num.toString(10);
          }
        }
        entity = entity.replace(/^0+/, "");
        if (isNaN(num) || numStr.toLowerCase() !== entity) {
          strictFail(parser, "Invalid character entity");
          return "&" + parser.entity + ";";
        }
        return String.fromCodePoint(num);
      }
      function beginWhiteSpace(parser, c) {
        if (c === "<") {
          parser.state = S.OPEN_WAKA;
          parser.startTagPosition = parser.position;
        } else if (!isWhitespace(c)) {
          strictFail(parser, "Non-whitespace before first tag.");
          parser.textNode = c;
          parser.state = S.TEXT;
        }
      }
      function charAt(chunk, i) {
        var result = "";
        if (i < chunk.length) {
          result = chunk.charAt(i);
        }
        return result;
      }
      function write(chunk) {
        var parser = this;
        if (this.error) {
          throw this.error;
        }
        if (parser.closed) {
          return error(
            parser,
            "Cannot write after close. Assign an onready handler."
          );
        }
        if (chunk === null) {
          return end(parser);
        }
        if (typeof chunk === "object") {
          chunk = chunk.toString();
        }
        var i = 0;
        var c = "";
        while (true) {
          c = charAt(chunk, i++);
          parser.c = c;
          if (!c) {
            break;
          }
          if (parser.trackPosition) {
            parser.position++;
            if (c === "\n") {
              parser.line++;
              parser.column = 0;
            } else {
              parser.column++;
            }
          }
          switch (parser.state) {
            case S.BEGIN:
              parser.state = S.BEGIN_WHITESPACE;
              if (c === "\uFEFF") {
                continue;
              }
              beginWhiteSpace(parser, c);
              continue;
            case S.BEGIN_WHITESPACE:
              beginWhiteSpace(parser, c);
              continue;
            case S.TEXT:
              if (parser.sawRoot && !parser.closedRoot) {
                var starti = i - 1;
                while (c && c !== "<" && c !== "&") {
                  c = charAt(chunk, i++);
                  if (c && parser.trackPosition) {
                    parser.position++;
                    if (c === "\n") {
                      parser.line++;
                      parser.column = 0;
                    } else {
                      parser.column++;
                    }
                  }
                }
                parser.textNode += chunk.substring(starti, i - 1);
              }
              if (c === "<" && !(parser.sawRoot && parser.closedRoot && !parser.strict)) {
                parser.state = S.OPEN_WAKA;
                parser.startTagPosition = parser.position;
              } else {
                if (!isWhitespace(c) && (!parser.sawRoot || parser.closedRoot)) {
                  strictFail(parser, "Text data outside of root node.");
                }
                if (c === "&") {
                  parser.state = S.TEXT_ENTITY;
                } else {
                  parser.textNode += c;
                }
              }
              continue;
            case S.SCRIPT:
              if (c === "<") {
                parser.state = S.SCRIPT_ENDING;
              } else {
                parser.script += c;
              }
              continue;
            case S.SCRIPT_ENDING:
              if (c === "/") {
                parser.state = S.CLOSE_TAG;
              } else {
                parser.script += "<" + c;
                parser.state = S.SCRIPT;
              }
              continue;
            case S.OPEN_WAKA:
              if (c === "!") {
                parser.state = S.SGML_DECL;
                parser.sgmlDecl = "";
              } else if (isWhitespace(c)) {
              } else if (isMatch(nameStart, c)) {
                parser.state = S.OPEN_TAG;
                parser.tagName = c;
              } else if (c === "/") {
                parser.state = S.CLOSE_TAG;
                parser.tagName = "";
              } else if (c === "?") {
                parser.state = S.PROC_INST;
                parser.procInstName = parser.procInstBody = "";
              } else {
                strictFail(parser, "Unencoded <");
                if (parser.startTagPosition + 1 < parser.position) {
                  var pad = parser.position - parser.startTagPosition;
                  c = new Array(pad).join(" ") + c;
                }
                parser.textNode += "<" + c;
                parser.state = S.TEXT;
              }
              continue;
            case S.SGML_DECL:
              if ((parser.sgmlDecl + c).toUpperCase() === CDATA) {
                emitNode(parser, "onopencdata");
                parser.state = S.CDATA;
                parser.sgmlDecl = "";
                parser.cdata = "";
              } else if (parser.sgmlDecl + c === "--") {
                parser.state = S.COMMENT;
                parser.comment = "";
                parser.sgmlDecl = "";
              } else if ((parser.sgmlDecl + c).toUpperCase() === DOCTYPE) {
                parser.state = S.DOCTYPE;
                if (parser.doctype || parser.sawRoot) {
                  strictFail(
                    parser,
                    "Inappropriately located doctype declaration"
                  );
                }
                parser.doctype = "";
                parser.sgmlDecl = "";
              } else if (c === ">") {
                emitNode(parser, "onsgmldeclaration", parser.sgmlDecl);
                parser.sgmlDecl = "";
                parser.state = S.TEXT;
              } else if (isQuote(c)) {
                parser.state = S.SGML_DECL_QUOTED;
                parser.sgmlDecl += c;
              } else {
                parser.sgmlDecl += c;
              }
              continue;
            case S.SGML_DECL_QUOTED:
              if (c === parser.q) {
                parser.state = S.SGML_DECL;
                parser.q = "";
              }
              parser.sgmlDecl += c;
              continue;
            case S.DOCTYPE:
              if (c === ">") {
                parser.state = S.TEXT;
                emitNode(parser, "ondoctype", parser.doctype);
                parser.doctype = true;
              } else {
                parser.doctype += c;
                if (c === "[") {
                  parser.state = S.DOCTYPE_DTD;
                } else if (isQuote(c)) {
                  parser.state = S.DOCTYPE_QUOTED;
                  parser.q = c;
                }
              }
              continue;
            case S.DOCTYPE_QUOTED:
              parser.doctype += c;
              if (c === parser.q) {
                parser.q = "";
                parser.state = S.DOCTYPE;
              }
              continue;
            case S.DOCTYPE_DTD:
              parser.doctype += c;
              if (c === "]") {
                parser.state = S.DOCTYPE;
              } else if (isQuote(c)) {
                parser.state = S.DOCTYPE_DTD_QUOTED;
                parser.q = c;
              }
              continue;
            case S.DOCTYPE_DTD_QUOTED:
              parser.doctype += c;
              if (c === parser.q) {
                parser.state = S.DOCTYPE_DTD;
                parser.q = "";
              }
              continue;
            case S.COMMENT:
              if (c === "-") {
                parser.state = S.COMMENT_ENDING;
              } else {
                parser.comment += c;
              }
              continue;
            case S.COMMENT_ENDING:
              if (c === "-") {
                parser.state = S.COMMENT_ENDED;
                parser.comment = textopts(parser.opt, parser.comment);
                if (parser.comment) {
                  emitNode(parser, "oncomment", parser.comment);
                }
                parser.comment = "";
              } else {
                parser.comment += "-" + c;
                parser.state = S.COMMENT;
              }
              continue;
            case S.COMMENT_ENDED:
              if (c !== ">") {
                strictFail(parser, "Malformed comment");
                parser.comment += "--" + c;
                parser.state = S.COMMENT;
              } else {
                parser.state = S.TEXT;
              }
              continue;
            case S.CDATA:
              if (c === "]") {
                parser.state = S.CDATA_ENDING;
              } else {
                parser.cdata += c;
              }
              continue;
            case S.CDATA_ENDING:
              if (c === "]") {
                parser.state = S.CDATA_ENDING_2;
              } else {
                parser.cdata += "]" + c;
                parser.state = S.CDATA;
              }
              continue;
            case S.CDATA_ENDING_2:
              if (c === ">") {
                if (parser.cdata) {
                  emitNode(parser, "oncdata", parser.cdata);
                }
                emitNode(parser, "onclosecdata");
                parser.cdata = "";
                parser.state = S.TEXT;
              } else if (c === "]") {
                parser.cdata += "]";
              } else {
                parser.cdata += "]]" + c;
                parser.state = S.CDATA;
              }
              continue;
            case S.PROC_INST:
              if (c === "?") {
                parser.state = S.PROC_INST_ENDING;
              } else if (isWhitespace(c)) {
                parser.state = S.PROC_INST_BODY;
              } else {
                parser.procInstName += c;
              }
              continue;
            case S.PROC_INST_BODY:
              if (!parser.procInstBody && isWhitespace(c)) {
                continue;
              } else if (c === "?") {
                parser.state = S.PROC_INST_ENDING;
              } else {
                parser.procInstBody += c;
              }
              continue;
            case S.PROC_INST_ENDING:
              if (c === ">") {
                emitNode(parser, "onprocessinginstruction", {
                  name: parser.procInstName,
                  body: parser.procInstBody
                });
                parser.procInstName = parser.procInstBody = "";
                parser.state = S.TEXT;
              } else {
                parser.procInstBody += "?" + c;
                parser.state = S.PROC_INST_BODY;
              }
              continue;
            case S.OPEN_TAG:
              if (isMatch(nameBody, c)) {
                parser.tagName += c;
              } else {
                newTag(parser);
                if (c === ">") {
                  openTag(parser);
                } else if (c === "/") {
                  parser.state = S.OPEN_TAG_SLASH;
                } else {
                  if (!isWhitespace(c)) {
                    strictFail(parser, "Invalid character in tag name");
                  }
                  parser.state = S.ATTRIB;
                }
              }
              continue;
            case S.OPEN_TAG_SLASH:
              if (c === ">") {
                openTag(parser, true);
                closeTag(parser);
              } else {
                strictFail(parser, "Forward-slash in opening tag not followed by >");
                parser.state = S.ATTRIB;
              }
              continue;
            case S.ATTRIB:
              if (isWhitespace(c)) {
                continue;
              } else if (c === ">") {
                openTag(parser);
              } else if (c === "/") {
                parser.state = S.OPEN_TAG_SLASH;
              } else if (isMatch(nameStart, c)) {
                parser.attribName = c;
                parser.attribValue = "";
                parser.state = S.ATTRIB_NAME;
              } else {
                strictFail(parser, "Invalid attribute name");
              }
              continue;
            case S.ATTRIB_NAME:
              if (c === "=") {
                parser.state = S.ATTRIB_VALUE;
              } else if (c === ">") {
                strictFail(parser, "Attribute without value");
                parser.attribValue = parser.attribName;
                attrib(parser);
                openTag(parser);
              } else if (isWhitespace(c)) {
                parser.state = S.ATTRIB_NAME_SAW_WHITE;
              } else if (isMatch(nameBody, c)) {
                parser.attribName += c;
              } else {
                strictFail(parser, "Invalid attribute name");
              }
              continue;
            case S.ATTRIB_NAME_SAW_WHITE:
              if (c === "=") {
                parser.state = S.ATTRIB_VALUE;
              } else if (isWhitespace(c)) {
                continue;
              } else {
                strictFail(parser, "Attribute without value");
                parser.tag.attributes[parser.attribName] = "";
                parser.attribValue = "";
                emitNode(parser, "onattribute", {
                  name: parser.attribName,
                  value: ""
                });
                parser.attribName = "";
                if (c === ">") {
                  openTag(parser);
                } else if (isMatch(nameStart, c)) {
                  parser.attribName = c;
                  parser.state = S.ATTRIB_NAME;
                } else {
                  strictFail(parser, "Invalid attribute name");
                  parser.state = S.ATTRIB;
                }
              }
              continue;
            case S.ATTRIB_VALUE:
              if (isWhitespace(c)) {
                continue;
              } else if (isQuote(c)) {
                parser.q = c;
                parser.state = S.ATTRIB_VALUE_QUOTED;
              } else {
                strictFail(parser, "Unquoted attribute value");
                parser.state = S.ATTRIB_VALUE_UNQUOTED;
                parser.attribValue = c;
              }
              continue;
            case S.ATTRIB_VALUE_QUOTED:
              if (c !== parser.q) {
                if (c === "&") {
                  parser.state = S.ATTRIB_VALUE_ENTITY_Q;
                } else {
                  parser.attribValue += c;
                }
                continue;
              }
              attrib(parser);
              parser.q = "";
              parser.state = S.ATTRIB_VALUE_CLOSED;
              continue;
            case S.ATTRIB_VALUE_CLOSED:
              if (isWhitespace(c)) {
                parser.state = S.ATTRIB;
              } else if (c === ">") {
                openTag(parser);
              } else if (c === "/") {
                parser.state = S.OPEN_TAG_SLASH;
              } else if (isMatch(nameStart, c)) {
                strictFail(parser, "No whitespace between attributes");
                parser.attribName = c;
                parser.attribValue = "";
                parser.state = S.ATTRIB_NAME;
              } else {
                strictFail(parser, "Invalid attribute name");
              }
              continue;
            case S.ATTRIB_VALUE_UNQUOTED:
              if (!isAttribEnd(c)) {
                if (c === "&") {
                  parser.state = S.ATTRIB_VALUE_ENTITY_U;
                } else {
                  parser.attribValue += c;
                }
                continue;
              }
              attrib(parser);
              if (c === ">") {
                openTag(parser);
              } else {
                parser.state = S.ATTRIB;
              }
              continue;
            case S.CLOSE_TAG:
              if (!parser.tagName) {
                if (isWhitespace(c)) {
                  continue;
                } else if (notMatch(nameStart, c)) {
                  if (parser.script) {
                    parser.script += "</" + c;
                    parser.state = S.SCRIPT;
                  } else {
                    strictFail(parser, "Invalid tagname in closing tag.");
                  }
                } else {
                  parser.tagName = c;
                }
              } else if (c === ">") {
                closeTag(parser);
              } else if (isMatch(nameBody, c)) {
                parser.tagName += c;
              } else if (parser.script) {
                parser.script += "</" + parser.tagName;
                parser.tagName = "";
                parser.state = S.SCRIPT;
              } else {
                if (!isWhitespace(c)) {
                  strictFail(parser, "Invalid tagname in closing tag");
                }
                parser.state = S.CLOSE_TAG_SAW_WHITE;
              }
              continue;
            case S.CLOSE_TAG_SAW_WHITE:
              if (isWhitespace(c)) {
                continue;
              }
              if (c === ">") {
                closeTag(parser);
              } else {
                strictFail(parser, "Invalid characters in closing tag");
              }
              continue;
            case S.TEXT_ENTITY:
            case S.ATTRIB_VALUE_ENTITY_Q:
            case S.ATTRIB_VALUE_ENTITY_U:
              var returnState;
              var buffer;
              switch (parser.state) {
                case S.TEXT_ENTITY:
                  returnState = S.TEXT;
                  buffer = "textNode";
                  break;
                case S.ATTRIB_VALUE_ENTITY_Q:
                  returnState = S.ATTRIB_VALUE_QUOTED;
                  buffer = "attribValue";
                  break;
                case S.ATTRIB_VALUE_ENTITY_U:
                  returnState = S.ATTRIB_VALUE_UNQUOTED;
                  buffer = "attribValue";
                  break;
              }
              if (c === ";") {
                parser[buffer] += parseEntity(parser);
                parser.entity = "";
                parser.state = returnState;
              } else if (isMatch(parser.entity.length ? entityBody : entityStart, c)) {
                parser.entity += c;
              } else {
                strictFail(parser, "Invalid character in entity name");
                parser[buffer] += "&" + parser.entity + c;
                parser.entity = "";
                parser.state = returnState;
              }
              continue;
            default:
              throw new Error(parser, "Unknown state: " + parser.state);
          }
        }
        if (parser.position >= parser.bufferCheckPosition) {
          checkBufferLength(parser);
        }
        return parser;
      }
      if (!String.fromCodePoint) {
        (function() {
          var stringFromCharCode = String.fromCharCode;
          var floor = Math.floor;
          var fromCodePoint = function() {
            var MAX_SIZE = 16384;
            var codeUnits = [];
            var highSurrogate;
            var lowSurrogate;
            var index = -1;
            var length = arguments.length;
            if (!length) {
              return "";
            }
            var result = "";
            while (++index < length) {
              var codePoint = Number(arguments[index]);
              if (!isFinite(codePoint) || codePoint < 0 || codePoint > 1114111 || floor(codePoint) !== codePoint) {
                throw RangeError("Invalid code point: " + codePoint);
              }
              if (codePoint <= 65535) {
                codeUnits.push(codePoint);
              } else {
                codePoint -= 65536;
                highSurrogate = (codePoint >> 10) + 55296;
                lowSurrogate = codePoint % 1024 + 56320;
                codeUnits.push(highSurrogate, lowSurrogate);
              }
              if (index + 1 === length || codeUnits.length > MAX_SIZE) {
                result += stringFromCharCode.apply(null, codeUnits);
                codeUnits.length = 0;
              }
            }
            return result;
          };
          if (Object.defineProperty) {
            Object.defineProperty(String, "fromCodePoint", {
              value: fromCodePoint,
              configurable: true,
              writable: true
            });
          } else {
            String.fromCodePoint = fromCodePoint;
          }
        })();
      }
    })(typeof exports === "undefined" ? exports.sax = {} : exports);
  }
});

// node_modules/.pnpm/sitemap@7.1.1/node_modules/sitemap/dist/lib/sitemap-parser.js
var require_sitemap_parser = __commonJS({
  "node_modules/.pnpm/sitemap@7.1.1/node_modules/sitemap/dist/lib/sitemap-parser.js"(exports) {
    "use strict";
    var __importDefault = exports && exports.__importDefault || function(mod) {
      return mod && mod.__esModule ? mod : { "default": mod };
    };
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.ObjectStreamToJSON = exports.parseSitemap = exports.XMLToSitemapItemStream = void 0;
    var sax_1 = __importDefault(require_sax());
    var stream_1 = require("stream");
    var types_1 = require_types();
    function isValidTagName(tagName) {
      return tagName in types_1.TagNames;
    }
    function tagTemplate() {
      return {
        img: [],
        video: [],
        links: [],
        url: ""
      };
    }
    function videoTemplate() {
      return {
        tag: [],
        thumbnail_loc: "",
        title: "",
        description: ""
      };
    }
    var imageTemplate = {
      url: ""
    };
    var linkTemplate = {
      lang: "",
      url: ""
    };
    function newsTemplate() {
      return {
        publication: { name: "", language: "" },
        publication_date: "",
        title: ""
      };
    }
    var defaultLogger = (level, ...message) => console[level](...message);
    var defaultStreamOpts = {
      logger: defaultLogger
    };
    var XMLToSitemapItemStream = class extends stream_1.Transform {
      constructor(opts = defaultStreamOpts) {
        var _a;
        opts.objectMode = true;
        super(opts);
        this.error = null;
        this.saxStream = sax_1.default.createStream(true, {
          xmlns: true,
          strictEntities: true,
          trim: true
        });
        this.level = opts.level || types_1.ErrorLevel.WARN;
        if (this.level !== types_1.ErrorLevel.SILENT && opts.logger !== false) {
          this.logger = (_a = opts.logger) !== null && _a !== void 0 ? _a : defaultLogger;
        } else {
          this.logger = () => void 0;
        }
        let currentItem = tagTemplate();
        let currentTag;
        let currentVideo = videoTemplate();
        let currentImage = { ...imageTemplate };
        let currentLink = { ...linkTemplate };
        let dontpushCurrentLink = false;
        this.saxStream.on("opentagstart", (tag) => {
          currentTag = tag.name;
          if (currentTag.startsWith("news:") && !currentItem.news) {
            currentItem.news = newsTemplate();
          }
        });
        this.saxStream.on("opentag", (tag) => {
          if (isValidTagName(tag.name)) {
            if (tag.name === "xhtml:link") {
              if (typeof tag.attributes.rel === "string" || typeof tag.attributes.href === "string") {
                return;
              }
              if (tag.attributes.rel.value === "alternate" && tag.attributes.hreflang) {
                currentLink.url = tag.attributes.href.value;
                if (typeof tag.attributes.hreflang === "string")
                  return;
                currentLink.lang = tag.attributes.hreflang.value;
              } else if (tag.attributes.rel.value === "alternate") {
                dontpushCurrentLink = true;
                currentItem.androidLink = tag.attributes.href.value;
              } else if (tag.attributes.rel.value === "amphtml") {
                dontpushCurrentLink = true;
                currentItem.ampLink = tag.attributes.href.value;
              } else {
                this.logger("log", "unhandled attr for xhtml:link", tag.attributes);
                this.err(`unhandled attr for xhtml:link ${tag.attributes}`);
              }
            }
          } else {
            this.logger("warn", "unhandled tag", tag.name);
            this.err(`unhandled tag: ${tag.name}`);
          }
        });
        this.saxStream.on("text", (text) => {
          switch (currentTag) {
            case "mobile:mobile":
              break;
            case types_1.TagNames.loc:
              currentItem.url = text;
              break;
            case types_1.TagNames.changefreq:
              if ((0, types_1.isValidChangeFreq)(text)) {
                currentItem.changefreq = text;
              }
              break;
            case types_1.TagNames.priority:
              currentItem.priority = parseFloat(text);
              break;
            case types_1.TagNames.lastmod:
              currentItem.lastmod = text;
              break;
            case types_1.TagNames["video:thumbnail_loc"]:
              currentVideo.thumbnail_loc = text;
              break;
            case types_1.TagNames["video:tag"]:
              currentVideo.tag.push(text);
              break;
            case types_1.TagNames["video:duration"]:
              currentVideo.duration = parseInt(text, 10);
              break;
            case types_1.TagNames["video:player_loc"]:
              currentVideo.player_loc = text;
              break;
            case types_1.TagNames["video:content_loc"]:
              currentVideo.content_loc = text;
              break;
            case types_1.TagNames["video:requires_subscription"]:
              if ((0, types_1.isValidYesNo)(text)) {
                currentVideo.requires_subscription = text;
              }
              break;
            case types_1.TagNames["video:publication_date"]:
              currentVideo.publication_date = text;
              break;
            case types_1.TagNames["video:id"]:
              currentVideo.id = text;
              break;
            case types_1.TagNames["video:restriction"]:
              currentVideo.restriction = text;
              break;
            case types_1.TagNames["video:view_count"]:
              currentVideo.view_count = parseInt(text, 10);
              break;
            case types_1.TagNames["video:uploader"]:
              currentVideo.uploader = text;
              break;
            case types_1.TagNames["video:family_friendly"]:
              if ((0, types_1.isValidYesNo)(text)) {
                currentVideo.family_friendly = text;
              }
              break;
            case types_1.TagNames["video:expiration_date"]:
              currentVideo.expiration_date = text;
              break;
            case types_1.TagNames["video:platform"]:
              currentVideo.platform = text;
              break;
            case types_1.TagNames["video:price"]:
              currentVideo.price = text;
              break;
            case types_1.TagNames["video:rating"]:
              currentVideo.rating = parseFloat(text);
              break;
            case types_1.TagNames["video:category"]:
              currentVideo.category = text;
              break;
            case types_1.TagNames["video:live"]:
              if ((0, types_1.isValidYesNo)(text)) {
                currentVideo.live = text;
              }
              break;
            case types_1.TagNames["video:gallery_loc"]:
              currentVideo.gallery_loc = text;
              break;
            case types_1.TagNames["image:loc"]:
              currentImage.url = text;
              break;
            case types_1.TagNames["image:geo_location"]:
              currentImage.geoLocation = text;
              break;
            case types_1.TagNames["image:license"]:
              currentImage.license = text;
              break;
            case types_1.TagNames["news:access"]:
              if (!currentItem.news) {
                currentItem.news = newsTemplate();
              }
              currentItem.news.access = text;
              break;
            case types_1.TagNames["news:genres"]:
              if (!currentItem.news) {
                currentItem.news = newsTemplate();
              }
              currentItem.news.genres = text;
              break;
            case types_1.TagNames["news:publication_date"]:
              if (!currentItem.news) {
                currentItem.news = newsTemplate();
              }
              currentItem.news.publication_date = text;
              break;
            case types_1.TagNames["news:keywords"]:
              if (!currentItem.news) {
                currentItem.news = newsTemplate();
              }
              currentItem.news.keywords = text;
              break;
            case types_1.TagNames["news:stock_tickers"]:
              if (!currentItem.news) {
                currentItem.news = newsTemplate();
              }
              currentItem.news.stock_tickers = text;
              break;
            case types_1.TagNames["news:language"]:
              if (!currentItem.news) {
                currentItem.news = newsTemplate();
              }
              currentItem.news.publication.language = text;
              break;
            case types_1.TagNames["video:title"]:
              currentVideo.title += text;
              break;
            case types_1.TagNames["video:description"]:
              currentVideo.description += text;
              break;
            case types_1.TagNames["news:name"]:
              if (!currentItem.news) {
                currentItem.news = newsTemplate();
              }
              currentItem.news.publication.name += text;
              break;
            case types_1.TagNames["news:title"]:
              if (!currentItem.news) {
                currentItem.news = newsTemplate();
              }
              currentItem.news.title += text;
              break;
            case types_1.TagNames["image:caption"]:
              if (!currentImage.caption) {
                currentImage.caption = text;
              } else {
                currentImage.caption += text;
              }
              break;
            case types_1.TagNames["image:title"]:
              if (!currentImage.title) {
                currentImage.title = text;
              } else {
                currentImage.title += text;
              }
              break;
            default:
              this.logger("log", "unhandled text for tag:", currentTag, `'${text}'`);
              this.err(`unhandled text for tag: ${currentTag} '${text}'`);
              break;
          }
        });
        this.saxStream.on("cdata", (text) => {
          switch (currentTag) {
            case types_1.TagNames["video:title"]:
              currentVideo.title += text;
              break;
            case types_1.TagNames["video:description"]:
              currentVideo.description += text;
              break;
            case types_1.TagNames["news:name"]:
              if (!currentItem.news) {
                currentItem.news = newsTemplate();
              }
              currentItem.news.publication.name += text;
              break;
            case types_1.TagNames["news:title"]:
              if (!currentItem.news) {
                currentItem.news = newsTemplate();
              }
              currentItem.news.title += text;
              break;
            case types_1.TagNames["image:caption"]:
              if (!currentImage.caption) {
                currentImage.caption = text;
              } else {
                currentImage.caption += text;
              }
              break;
            case types_1.TagNames["image:title"]:
              if (!currentImage.title) {
                currentImage.title = text;
              } else {
                currentImage.title += text;
              }
              break;
            default:
              this.logger("log", "unhandled cdata for tag:", currentTag);
              this.err(`unhandled cdata for tag: ${currentTag}`);
              break;
          }
        });
        this.saxStream.on("attribute", (attr) => {
          switch (currentTag) {
            case types_1.TagNames["urlset"]:
            case types_1.TagNames["xhtml:link"]:
            case types_1.TagNames["video:id"]:
              break;
            case types_1.TagNames["video:restriction"]:
              if (attr.name === "relationship" && (0, types_1.isAllowDeny)(attr.value)) {
                currentVideo["restriction:relationship"] = attr.value;
              } else {
                this.logger("log", "unhandled attr", currentTag, attr.name);
                this.err(`unhandled attr: ${currentTag} ${attr.name}`);
              }
              break;
            case types_1.TagNames["video:price"]:
              if (attr.name === "type" && (0, types_1.isPriceType)(attr.value)) {
                currentVideo["price:type"] = attr.value;
              } else if (attr.name === "currency") {
                currentVideo["price:currency"] = attr.value;
              } else if (attr.name === "resolution" && (0, types_1.isResolution)(attr.value)) {
                currentVideo["price:resolution"] = attr.value;
              } else {
                this.logger("log", "unhandled attr for video:price", attr.name);
                this.err(`unhandled attr: ${currentTag} ${attr.name}`);
              }
              break;
            case types_1.TagNames["video:player_loc"]:
              if (attr.name === "autoplay") {
                currentVideo["player_loc:autoplay"] = attr.value;
              } else if (attr.name === "allow_embed" && (0, types_1.isValidYesNo)(attr.value)) {
                currentVideo["player_loc:allow_embed"] = attr.value;
              } else {
                this.logger("log", "unhandled attr for video:player_loc", attr.name);
                this.err(`unhandled attr: ${currentTag} ${attr.name}`);
              }
              break;
            case types_1.TagNames["video:platform"]:
              if (attr.name === "relationship" && (0, types_1.isAllowDeny)(attr.value)) {
                currentVideo["platform:relationship"] = attr.value;
              } else {
                this.logger("log", "unhandled attr for video:platform", attr.name, attr.value);
                this.err(`unhandled attr: ${currentTag} ${attr.name} ${attr.value}`);
              }
              break;
            case types_1.TagNames["video:gallery_loc"]:
              if (attr.name === "title") {
                currentVideo["gallery_loc:title"] = attr.value;
              } else {
                this.logger("log", "unhandled attr for video:galler_loc", attr.name);
                this.err(`unhandled attr: ${currentTag} ${attr.name}`);
              }
              break;
            case types_1.TagNames["video:uploader"]:
              if (attr.name === "info") {
                currentVideo["uploader:info"] = attr.value;
              } else {
                this.logger("log", "unhandled attr for video:uploader", attr.name);
                this.err(`unhandled attr: ${currentTag} ${attr.name}`);
              }
              break;
            default:
              this.logger("log", "unhandled attr", currentTag, attr.name);
              this.err(`unhandled attr: ${currentTag} ${attr.name}`);
          }
        });
        this.saxStream.on("closetag", (tag) => {
          switch (tag) {
            case types_1.TagNames.url:
              this.push(currentItem);
              currentItem = tagTemplate();
              break;
            case types_1.TagNames["video:video"]:
              currentItem.video.push(currentVideo);
              currentVideo = videoTemplate();
              break;
            case types_1.TagNames["image:image"]:
              currentItem.img.push(currentImage);
              currentImage = { ...imageTemplate };
              break;
            case types_1.TagNames["xhtml:link"]:
              if (!dontpushCurrentLink) {
                currentItem.links.push(currentLink);
              }
              currentLink = { ...linkTemplate };
              break;
            default:
              break;
          }
        });
      }
      _transform(data, encoding, callback) {
        try {
          this.saxStream.write(data, encoding);
          callback(this.level === types_1.ErrorLevel.THROW ? this.error : null);
        } catch (error) {
          callback(error);
        }
      }
      err(msg) {
        if (!this.error)
          this.error = new Error(msg);
      }
    };
    exports.XMLToSitemapItemStream = XMLToSitemapItemStream;
    async function parseSitemap(xml) {
      const urls = [];
      return new Promise((resolve2, reject) => {
        xml.pipe(new XMLToSitemapItemStream()).on("data", (smi) => urls.push(smi)).on("end", () => {
          resolve2(urls);
        }).on("error", (error) => {
          reject(error);
        });
      });
    }
    exports.parseSitemap = parseSitemap;
    var defaultObjectStreamOpts = {
      lineSeparated: false
    };
    var ObjectStreamToJSON = class extends stream_1.Transform {
      constructor(opts = defaultObjectStreamOpts) {
        opts.writableObjectMode = true;
        super(opts);
        this.lineSeparated = opts.lineSeparated;
        this.firstWritten = false;
      }
      _transform(chunk, encoding, cb) {
        if (!this.firstWritten) {
          this.firstWritten = true;
          if (!this.lineSeparated) {
            this.push("[");
          }
        } else if (this.lineSeparated) {
          this.push("\n");
        } else {
          this.push(",");
        }
        if (chunk) {
          this.push(JSON.stringify(chunk));
        }
        cb();
      }
      _flush(cb) {
        if (!this.lineSeparated) {
          this.push("]");
        }
        cb();
      }
    };
    exports.ObjectStreamToJSON = ObjectStreamToJSON;
  }
});

// node_modules/.pnpm/sitemap@7.1.1/node_modules/sitemap/dist/lib/sitemap-index-parser.js
var require_sitemap_index_parser = __commonJS({
  "node_modules/.pnpm/sitemap@7.1.1/node_modules/sitemap/dist/lib/sitemap-index-parser.js"(exports) {
    "use strict";
    var __importDefault = exports && exports.__importDefault || function(mod) {
      return mod && mod.__esModule ? mod : { "default": mod };
    };
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.IndexObjectStreamToJSON = exports.parseSitemapIndex = exports.XMLToSitemapIndexStream = void 0;
    var sax_1 = __importDefault(require_sax());
    var stream_1 = require("stream");
    var types_1 = require_types();
    function isValidTagName(tagName) {
      return tagName in types_1.IndexTagNames;
    }
    function tagTemplate() {
      return {
        url: ""
      };
    }
    var defaultLogger = (level, ...message) => console[level](...message);
    var defaultStreamOpts = {
      logger: defaultLogger
    };
    var XMLToSitemapIndexStream = class extends stream_1.Transform {
      constructor(opts = defaultStreamOpts) {
        var _a;
        opts.objectMode = true;
        super(opts);
        this.saxStream = sax_1.default.createStream(true, {
          xmlns: true,
          strictEntities: true,
          trim: true
        });
        this.level = opts.level || types_1.ErrorLevel.WARN;
        if (this.level !== types_1.ErrorLevel.SILENT && opts.logger !== false) {
          this.logger = (_a = opts.logger) !== null && _a !== void 0 ? _a : defaultLogger;
        } else {
          this.logger = () => void 0;
        }
        let currentItem = tagTemplate();
        let currentTag;
        this.saxStream.on("opentagstart", (tag) => {
          currentTag = tag.name;
        });
        this.saxStream.on("opentag", (tag) => {
          if (!isValidTagName(tag.name)) {
            this.logger("warn", "unhandled tag", tag.name);
          }
        });
        this.saxStream.on("text", (text) => {
          switch (currentTag) {
            case types_1.IndexTagNames.loc:
              currentItem.url = text;
              break;
            case types_1.IndexTagNames.lastmod:
              currentItem.lastmod = text;
              break;
            default:
              this.logger("log", "unhandled text for tag:", currentTag, `'${text}'`);
              break;
          }
        });
        this.saxStream.on("cdata", (_text) => {
          switch (currentTag) {
            default:
              this.logger("log", "unhandled cdata for tag:", currentTag);
              break;
          }
        });
        this.saxStream.on("attribute", (attr) => {
          switch (currentTag) {
            case types_1.IndexTagNames.sitemapindex:
              break;
            default:
              this.logger("log", "unhandled attr", currentTag, attr.name);
          }
        });
        this.saxStream.on("closetag", (tag) => {
          switch (tag) {
            case types_1.IndexTagNames.sitemap:
              this.push(currentItem);
              currentItem = tagTemplate();
              break;
            default:
              break;
          }
        });
      }
      _transform(data, encoding, callback) {
        try {
          this.saxStream.write(data, encoding);
          callback();
        } catch (error) {
          callback(error);
        }
      }
    };
    exports.XMLToSitemapIndexStream = XMLToSitemapIndexStream;
    async function parseSitemapIndex(xml) {
      const urls = [];
      return new Promise((resolve2, reject) => {
        xml.pipe(new XMLToSitemapIndexStream()).on("data", (smi) => urls.push(smi)).on("end", () => {
          resolve2(urls);
        }).on("error", (error) => {
          reject(error);
        });
      });
    }
    exports.parseSitemapIndex = parseSitemapIndex;
    var defaultObjectStreamOpts = {
      lineSeparated: false
    };
    var IndexObjectStreamToJSON = class extends stream_1.Transform {
      constructor(opts = defaultObjectStreamOpts) {
        opts.writableObjectMode = true;
        super(opts);
        this.lineSeparated = opts.lineSeparated;
        this.firstWritten = false;
      }
      _transform(chunk, encoding, cb) {
        if (!this.firstWritten) {
          this.firstWritten = true;
          if (!this.lineSeparated) {
            this.push("[");
          }
        } else if (this.lineSeparated) {
          this.push("\n");
        } else {
          this.push(",");
        }
        if (chunk) {
          this.push(JSON.stringify(chunk));
        }
        cb();
      }
      _flush(cb) {
        if (!this.lineSeparated) {
          this.push("]");
        }
        cb();
      }
    };
    exports.IndexObjectStreamToJSON = IndexObjectStreamToJSON;
  }
});

// node_modules/.pnpm/sitemap@7.1.1/node_modules/sitemap/dist/lib/sitemap-simple.js
var require_sitemap_simple = __commonJS({
  "node_modules/.pnpm/sitemap@7.1.1/node_modules/sitemap/dist/lib/sitemap-simple.js"(exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.simpleSitemapAndIndex = void 0;
    var index_1 = require_dist();
    var zlib_1 = require("zlib");
    var fs_1 = require("fs");
    var path_1 = require("path");
    var stream_1 = require("stream");
    var util_1 = require("util");
    var url_1 = require("url");
    var pipeline = (0, util_1.promisify)(stream_1.pipeline);
    var simpleSitemapAndIndex = async ({
      hostname,
      sitemapHostname = hostname,
      sourceData,
      destinationDir,
      limit = 5e4,
      gzip = true,
      publicBasePath = "./"
    }) => {
      await fs_1.promises.mkdir(destinationDir, { recursive: true });
      const sitemapAndIndexStream = new index_1.SitemapAndIndexStream({
        limit,
        getSitemapStream: (i) => {
          const sitemapStream = new index_1.SitemapStream({
            hostname
          });
          const path = `./sitemap-${i}.xml`;
          const writePath2 = (0, path_1.resolve)(destinationDir, path + (gzip ? ".gz" : ""));
          if (!publicBasePath.endsWith("/")) {
            publicBasePath += "/";
          }
          const publicPath = (0, path_1.normalize)(publicBasePath + path);
          let pipeline2;
          if (gzip) {
            pipeline2 = sitemapStream.pipe((0, zlib_1.createGzip)()).pipe((0, fs_1.createWriteStream)(writePath2));
          } else {
            pipeline2 = sitemapStream.pipe((0, fs_1.createWriteStream)(writePath2));
          }
          return [
            new url_1.URL(`${publicPath}${gzip ? ".gz" : ""}`, sitemapHostname).toString(),
            sitemapStream,
            pipeline2
          ];
        }
      });
      let src;
      if (typeof sourceData === "string") {
        src = (0, index_1.lineSeparatedURLsToSitemapOptions)((0, fs_1.createReadStream)(sourceData));
      } else if (sourceData instanceof stream_1.Readable) {
        src = sourceData;
      } else if (Array.isArray(sourceData)) {
        src = stream_1.Readable.from(sourceData);
      } else {
        throw new Error("unhandled source type. You've passed in data that is not supported");
      }
      const writePath = (0, path_1.resolve)(destinationDir, `./sitemap-index.xml${gzip ? ".gz" : ""}`);
      if (gzip) {
        return pipeline(src, sitemapAndIndexStream, (0, zlib_1.createGzip)(), (0, fs_1.createWriteStream)(writePath));
      } else {
        return pipeline(src, sitemapAndIndexStream, (0, fs_1.createWriteStream)(writePath));
      }
    };
    exports.simpleSitemapAndIndex = simpleSitemapAndIndex;
    exports.default = exports.simpleSitemapAndIndex;
  }
});

// node_modules/.pnpm/sitemap@7.1.1/node_modules/sitemap/dist/index.js
var require_dist = __commonJS({
  "node_modules/.pnpm/sitemap@7.1.1/node_modules/sitemap/dist/index.js"(exports) {
    "use strict";
    var __createBinding = exports && exports.__createBinding || (Object.create ? function(o, m, k, k2) {
      if (k2 === void 0)
        k2 = k;
      Object.defineProperty(o, k2, { enumerable: true, get: function() {
        return m[k];
      } });
    } : function(o, m, k, k2) {
      if (k2 === void 0)
        k2 = k;
      o[k2] = m[k];
    });
    var __exportStar = exports && exports.__exportStar || function(m, exports2) {
      for (var p in m)
        if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports2, p))
          __createBinding(exports2, m, p);
    };
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.simpleSitemapAndIndex = exports.IndexObjectStreamToJSON = exports.XMLToSitemapIndexStream = exports.parseSitemapIndex = exports.ObjectStreamToJSON = exports.XMLToSitemapItemStream = exports.parseSitemap = exports.xmlLint = exports.ReadlineStream = exports.normalizeURL = exports.validateSMIOptions = exports.mergeStreams = exports.lineSeparatedURLsToSitemapOptions = exports.SitemapStream = exports.streamToPromise = exports.SitemapAndIndexStream = exports.SitemapIndexStream = exports.IndexTagNames = exports.SitemapItemStream = void 0;
    var sitemap_item_stream_1 = require_sitemap_item_stream();
    Object.defineProperty(exports, "SitemapItemStream", { enumerable: true, get: function() {
      return sitemap_item_stream_1.SitemapItemStream;
    } });
    var sitemap_index_stream_1 = require_sitemap_index_stream();
    Object.defineProperty(exports, "IndexTagNames", { enumerable: true, get: function() {
      return sitemap_index_stream_1.IndexTagNames;
    } });
    Object.defineProperty(exports, "SitemapIndexStream", { enumerable: true, get: function() {
      return sitemap_index_stream_1.SitemapIndexStream;
    } });
    Object.defineProperty(exports, "SitemapAndIndexStream", { enumerable: true, get: function() {
      return sitemap_index_stream_1.SitemapAndIndexStream;
    } });
    var sitemap_stream_1 = require_sitemap_stream();
    Object.defineProperty(exports, "streamToPromise", { enumerable: true, get: function() {
      return sitemap_stream_1.streamToPromise;
    } });
    Object.defineProperty(exports, "SitemapStream", { enumerable: true, get: function() {
      return sitemap_stream_1.SitemapStream;
    } });
    __exportStar(require_errors(), exports);
    __exportStar(require_types(), exports);
    var utils_1 = require_utils();
    Object.defineProperty(exports, "lineSeparatedURLsToSitemapOptions", { enumerable: true, get: function() {
      return utils_1.lineSeparatedURLsToSitemapOptions;
    } });
    Object.defineProperty(exports, "mergeStreams", { enumerable: true, get: function() {
      return utils_1.mergeStreams;
    } });
    Object.defineProperty(exports, "validateSMIOptions", { enumerable: true, get: function() {
      return utils_1.validateSMIOptions;
    } });
    Object.defineProperty(exports, "normalizeURL", { enumerable: true, get: function() {
      return utils_1.normalizeURL;
    } });
    Object.defineProperty(exports, "ReadlineStream", { enumerable: true, get: function() {
      return utils_1.ReadlineStream;
    } });
    var xmllint_1 = require_xmllint();
    Object.defineProperty(exports, "xmlLint", { enumerable: true, get: function() {
      return xmllint_1.xmlLint;
    } });
    var sitemap_parser_1 = require_sitemap_parser();
    Object.defineProperty(exports, "parseSitemap", { enumerable: true, get: function() {
      return sitemap_parser_1.parseSitemap;
    } });
    Object.defineProperty(exports, "XMLToSitemapItemStream", { enumerable: true, get: function() {
      return sitemap_parser_1.XMLToSitemapItemStream;
    } });
    Object.defineProperty(exports, "ObjectStreamToJSON", { enumerable: true, get: function() {
      return sitemap_parser_1.ObjectStreamToJSON;
    } });
    var sitemap_index_parser_1 = require_sitemap_index_parser();
    Object.defineProperty(exports, "parseSitemapIndex", { enumerable: true, get: function() {
      return sitemap_index_parser_1.parseSitemapIndex;
    } });
    Object.defineProperty(exports, "XMLToSitemapIndexStream", { enumerable: true, get: function() {
      return sitemap_index_parser_1.XMLToSitemapIndexStream;
    } });
    Object.defineProperty(exports, "IndexObjectStreamToJSON", { enumerable: true, get: function() {
      return sitemap_index_parser_1.IndexObjectStreamToJSON;
    } });
    var sitemap_simple_1 = require_sitemap_simple();
    Object.defineProperty(exports, "simpleSitemapAndIndex", { enumerable: true, get: function() {
      return sitemap_simple_1.simpleSitemapAndIndex;
    } });
  }
});

// node_modules/.pnpm/xml-parser-xo@3.2.0/node_modules/xml-parser-xo/index.js
var require_xml_parser_xo = __commonJS({
  "node_modules/.pnpm/xml-parser-xo@3.2.0/node_modules/xml-parser-xo/index.js"(exports, module2) {
    function parse(xml, options = {}) {
      options.filter = options.filter || (() => true);
      function nextChild() {
        return tag() || content() || comment() || cdata();
      }
      function nextRootChild() {
        match(/\s*/);
        return tag(true) || comment() || doctype() || processingInstruction(false);
      }
      function document() {
        const decl = declaration();
        const children = [];
        let documentRootNode;
        let child = nextRootChild();
        while (child) {
          if (child.node.type === "Element") {
            if (documentRootNode) {
              throw new Error("Found multiple root nodes");
            }
            documentRootNode = child.node;
          }
          if (!child.excluded) {
            children.push(child.node);
          }
          child = nextRootChild();
        }
        if (!documentRootNode) {
          throw new Error("Failed to parse XML");
        }
        return {
          declaration: decl ? decl.node : null,
          root: documentRootNode,
          children
        };
      }
      function declaration() {
        return processingInstruction(true);
      }
      function processingInstruction(matchDeclaration) {
        const m = matchDeclaration ? match(/^<\?(xml)\s*/) : match(/^<\?([\w-:.]+)\s*/);
        if (!m)
          return;
        const node = {
          name: m[1],
          type: "ProcessingInstruction",
          attributes: {}
        };
        while (!(eos() || is("?>"))) {
          const attr = attribute();
          if (!attr)
            return node;
          node.attributes[attr.name] = attr.value;
        }
        match(/\?>/);
        return {
          excluded: matchDeclaration ? false : options.filter(node) === false,
          node
        };
      }
      function tag(matchRoot) {
        const m = match(/^<([\w-:.]+)\s*/);
        if (!m)
          return;
        const node = {
          type: "Element",
          name: m[1],
          attributes: {},
          children: []
        };
        while (!(eos() || is(">") || is("?>") || is("/>"))) {
          const attr = attribute();
          if (!attr)
            return node;
          node.attributes[attr.name] = attr.value;
        }
        const excluded = matchRoot ? false : options.filter(node) === false;
        if (match(/^\s*\/>/)) {
          node.children = null;
          return {
            excluded,
            node
          };
        }
        match(/\??>/);
        if (!excluded) {
          let child = nextChild();
          while (child) {
            if (!child.excluded) {
              node.children.push(child.node);
            }
            child = nextChild();
          }
        }
        match(/^<\/[\w-:.]+>/);
        return {
          excluded,
          node
        };
      }
      function doctype() {
        const m = match(/^<!DOCTYPE\s+[^>]*>/);
        if (m) {
          const node = {
            type: "DocumentType",
            content: m[0]
          };
          return {
            excluded: options.filter(node) === false,
            node
          };
        }
      }
      function cdata() {
        if (xml.startsWith("<![CDATA[")) {
          const endPositionStart = xml.indexOf("]]>");
          if (endPositionStart > -1) {
            const endPositionFinish = endPositionStart + 3;
            const node = {
              type: "CDATA",
              content: xml.substring(0, endPositionFinish)
            };
            xml = xml.slice(endPositionFinish);
            return {
              excluded: options.filter(node) === false,
              node
            };
          }
        }
      }
      function comment() {
        const m = match(/^<!--[\s\S]*?-->/);
        if (m) {
          const node = {
            type: "Comment",
            content: m[0]
          };
          return {
            excluded: options.filter(node) === false,
            node
          };
        }
      }
      function content() {
        const m = match(/^([^<]+)/);
        if (m) {
          const node = {
            type: "Text",
            content: m[1]
          };
          return {
            excluded: options.filter(node) === false,
            node
          };
        }
      }
      function attribute() {
        const m = match(/([\w-:.]+)\s*=\s*("[^"]*"|'[^']*'|\w+)\s*/);
        if (!m)
          return;
        return { name: m[1], value: strip(m[2]) };
      }
      function strip(val) {
        return val.replace(/^['"]|['"]$/g, "");
      }
      function match(re) {
        const m = xml.match(re);
        if (!m)
          return;
        xml = xml.slice(m[0].length);
        return m;
      }
      function eos() {
        return 0 === xml.length;
      }
      function is(prefix) {
        return 0 === xml.indexOf(prefix);
      }
      xml = xml.trim();
      return document();
    }
    module2.exports = parse;
  }
});

// node_modules/.pnpm/xml-formatter@2.6.1/node_modules/xml-formatter/index.js
var require_xml_formatter = __commonJS({
  "node_modules/.pnpm/xml-formatter@2.6.1/node_modules/xml-formatter/index.js"(exports, module2) {
    function newLine(state) {
      if (!state.options.indentation && !state.options.lineSeparator)
        return;
      state.content += state.options.lineSeparator;
      let i;
      for (i = 0; i < state.level; i++) {
        state.content += state.options.indentation;
      }
    }
    function appendContent(state, content) {
      state.content += content;
    }
    function processNode(node, state, preserveSpace) {
      if (typeof node.content === "string") {
        processContentNode(node, state, preserveSpace);
      } else if (node.type === "Element") {
        processElementNode(node, state, preserveSpace);
      } else if (node.type === "ProcessingInstruction") {
        processProcessingIntruction(node, state, preserveSpace);
      } else {
        throw new Error("Unknown node type: " + node.type);
      }
    }
    function processContentNode(node, state, preserveSpace) {
      if (!preserveSpace) {
        node.content = node.content.trim();
      }
      if (node.content.length > 0) {
        if (!preserveSpace && state.content.length > 0) {
          newLine(state);
        }
        appendContent(state, node.content);
      }
    }
    function processElementNode(node, state, preserveSpace) {
      if (!preserveSpace && state.content.length > 0) {
        newLine(state);
      }
      appendContent(state, "<" + node.name);
      processAttributes(state, node.attributes);
      if (node.children === null) {
        const selfClosingNodeClosingTag = state.options.whiteSpaceAtEndOfSelfclosingTag ? " />" : "/>";
        appendContent(state, selfClosingNodeClosingTag);
      } else if (node.children.length === 0) {
        appendContent(state, "></" + node.name + ">");
      } else {
        appendContent(state, ">");
        state.level++;
        let nodePreserveSpace = node.attributes["xml:space"] === "preserve";
        if (!nodePreserveSpace && state.options.collapseContent) {
          let containsTextNodes = false;
          let containsTextNodesWithLineBreaks = false;
          let containsNonTextNodes = false;
          node.children.forEach(function(child, index) {
            if (child.type === "Text") {
              if (child.content.includes("\n")) {
                containsTextNodesWithLineBreaks = true;
                child.content = child.content.trim();
              } else if (index === 0 || index === node.children.length - 1) {
                if (child.content.trim().length === 0) {
                  child.content = "";
                }
              }
              if (child.content.length > 0) {
                containsTextNodes = true;
              }
            } else if (child.type === "CDATA") {
              containsTextNodes = true;
            } else {
              containsNonTextNodes = true;
            }
          });
          if (containsTextNodes && (!containsNonTextNodes || !containsTextNodesWithLineBreaks)) {
            nodePreserveSpace = true;
          }
        }
        node.children.forEach(function(child) {
          processNode(child, state, preserveSpace || nodePreserveSpace, state.options);
        });
        state.level--;
        if (!preserveSpace && !nodePreserveSpace) {
          newLine(state);
        }
        appendContent(state, "</" + node.name + ">");
      }
    }
    function processAttributes(state, attributes) {
      Object.keys(attributes).forEach(function(attr) {
        const escaped = attributes[attr].replace(/"/g, "&quot;");
        appendContent(state, " " + attr + '="' + escaped + '"');
      });
    }
    function processProcessingIntruction(node, state) {
      if (state.content.length > 0) {
        newLine(state);
      }
      appendContent(state, "<?" + node.name);
      processAttributes(state, node.attributes);
      appendContent(state, "?>");
    }
    function format2(xml, options = {}) {
      options.indentation = "indentation" in options ? options.indentation : "    ";
      options.collapseContent = options.collapseContent === true;
      options.lineSeparator = "lineSeparator" in options ? options.lineSeparator : "\r\n";
      options.whiteSpaceAtEndOfSelfclosingTag = !!options.whiteSpaceAtEndOfSelfclosingTag;
      const parser = require_xml_parser_xo();
      const parsedXml = parser(xml, { filter: options.filter });
      const state = { content: "", level: 0, options };
      if (parsedXml.declaration) {
        processProcessingIntruction(parsedXml.declaration, state);
      }
      parsedXml.children.forEach(function(child) {
        processNode(child, state, false);
      });
      return state.content.replace(/\r\n/g, "\n").replace(/\n/g, options.lineSeparator);
    }
    module2.exports = format2;
  }
});

// src/index.ts
var src_exports = {};
__export(src_exports, {
  default: () => generateSitemap,
  flatRoutes: () => flatRoutes,
  getDestPath: () => getDestPath,
  getFinalSitemapPath: () => getFinalSitemapPath,
  getResolvedPath: () => getResolvedPath,
  getSitemapLinks: () => getSitemapLinks,
  writeRobotFile: () => writeRobotFile,
  writeXmlFile: () => writeXmlFile
});
module.exports = __toCommonJS(src_exports);
var import_fs = require("fs");
var import_path = require("path");
var import_sitemap = __toESM(require_dist());

// node_modules/.pnpm/@antfu+utils@0.5.2/node_modules/@antfu/utils/dist/index.mjs
function ensurePrefix(prefix, str) {
  if (!str.startsWith(prefix))
    return prefix + str;
  return str;
}
var Node = class {
  value;
  next;
  constructor(value) {
    this.value = value;
  }
};
var _head, _tail, _size;
var Queue = class {
  constructor() {
    __privateAdd(this, _head, void 0);
    __privateAdd(this, _tail, void 0);
    __privateAdd(this, _size, void 0);
    this.clear();
  }
  enqueue(value) {
    const node = new Node(value);
    if (__privateGet(this, _head)) {
      __privateGet(this, _tail).next = node;
      __privateSet(this, _tail, node);
    } else {
      __privateSet(this, _head, node);
      __privateSet(this, _tail, node);
    }
    __privateWrapper(this, _size)._++;
  }
  dequeue() {
    const current = __privateGet(this, _head);
    if (!current) {
      return;
    }
    __privateSet(this, _head, __privateGet(this, _head).next);
    __privateWrapper(this, _size)._--;
    return current.value;
  }
  clear() {
    __privateSet(this, _head, void 0);
    __privateSet(this, _tail, void 0);
    __privateSet(this, _size, 0);
  }
  get size() {
    return __privateGet(this, _size);
  }
  *[Symbol.iterator]() {
    let current = __privateGet(this, _head);
    while (current) {
      yield current.value;
      current = current.next;
    }
  }
};
_head = new WeakMap();
_tail = new WeakMap();
_size = new WeakMap();
var VOID = Symbol("p-void");

// src/index.ts
var import_xml_formatter = __toESM(require_xml_formatter());

// src/utils.ts
var dynamicRouteRE = /\[.+\]/;
var nuxtDynamicRouteRE = /_[\s\S]*/;
function isDynamicRoute(routePath, nuxtStyle) {
  if (!routePath)
    return false;
  return nuxtStyle ? nuxtDynamicRouteRE.test(routePath) : dynamicRouteRE.test(routePath);
}
function ensureSuffix(suffix, str) {
  if (!str.endsWith(suffix))
    return str + suffix;
  return str;
}
function removeMaybeSuffix(suffix, str) {
  if (!str.endsWith(suffix))
    return str;
  return str.slice(0, -suffix.length);
}

// src/options.ts
function resolveOptions(userOptions) {
  return Object.assign(
    {
      hostname: "http://localhost/",
      routes: [],
      exclude: [],
      filename: "sitemap",
      dest: "public",
      changefreq: "daily",
      priority: 1,
      lastmod: new Date(),
      readable: false,
      nuxtStyle: false,
      allowRobots: true
    },
    userOptions
  );
}

// src/index.ts
function generateSitemap(options) {
  const resolvedOptions = resolveOptions(options);
  const RESOLVED_PATH = getResolvedPath(resolvedOptions.filename, ".xml", resolvedOptions.dest);
  const RESOLVED_ROBOT_PATH = getResolvedPath("robots", ".txt", resolvedOptions.dest);
  if (!resolvedOptions.routes.length)
    return;
  if (!(0, import_fs.existsSync)(getDestPath(resolvedOptions.dest)))
    (0, import_fs.mkdirSync)(getDestPath(resolvedOptions.dest));
  const stream = new import_sitemap.SitemapStream();
  getSitemapLinks(resolvedOptions).forEach((item) => stream.write(item));
  (0, import_sitemap.streamToPromise)(stream).then((sitemap) => {
    writeXmlFile(RESOLVED_PATH, sitemap.toString("utf-8"), resolvedOptions);
  });
  stream.end();
  writeRobotFile(RESOLVED_ROBOT_PATH, resolvedOptions);
}
function getSitemapLinks(options) {
  const hostname = options.hostname;
  const routes = flatRoutes([...options.routes.values()]);
  return routes.filter((pageRoute) => !isDynamicRoute(pageRoute.component, options.nuxtStyle) || typeof pageRoute === "string").filter((pageRoute) => !options.exclude.includes(pageRoute.path)).map((pageRoute) => ({
    url: `${removeMaybeSuffix("/", hostname)}${typeof pageRoute === "string" ? pageRoute : pageRoute.path}`,
    changefreq: options.changefreq,
    priority: options.priority,
    lastmod: options.lastmod
  }));
}
function flatRoutes(routes) {
  const flatRoutes2 = /* @__PURE__ */ new Set();
  const getFlatRoutes = (routes2, prefix = "") => {
    prefix = prefix.replace(/\/$/g, "");
    for (const route of routes2) {
      if (typeof route === "string") {
        flatRoutes2.add({
          component: void 0,
          path: route
        });
      } else {
        let path = route.path;
        if (route.path) {
          path = prefix && !route.path.startsWith("/") ? `${prefix}/${route.path}` : route.path;
          flatRoutes2.add({
            component: route.component,
            path
          });
        }
        if (Array.isArray(route.children))
          getFlatRoutes(route.children, path);
      }
    }
  };
  getFlatRoutes(routes);
  return [...flatRoutes2];
}
function getDestPath(dest) {
  return ensurePrefix("./", dest);
}
function getResolvedPath(filename, extension, dest) {
  const filenameWithExtension = ensureSuffix(extension, filename);
  return (0, import_path.resolve)(`${getDestPath(dest)}/${filenameWithExtension}`);
}
function writeXmlFile(resolvedPath, str, options) {
  (0, import_fs.writeFileSync)(resolvedPath, options.readable ? (0, import_xml_formatter.default)(str) : str);
}
function writeRobotFile(resolvedPath, options) {
  const str = "User-agent: *\n".concat(`${options.allowRobots ? "Allow" : "Disallow"}: /

`).concat(`Sitemap: ${getFinalSitemapPath(options)}`);
  (0, import_fs.writeFileSync)(resolvedPath, str);
}
function getFinalSitemapPath(options) {
  return `${ensureSuffix("/", options.hostname)}${ensureSuffix(".xml", options.filename)}`;
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  flatRoutes,
  getDestPath,
  getFinalSitemapPath,
  getResolvedPath,
  getSitemapLinks,
  writeRobotFile,
  writeXmlFile
});
/*!
 * Sitemap
 * Copyright(c) 2011 Eugene Kalinin
 * MIT Licensed
 */
/*! http://mths.be/fromcodepoint v0.1.0 by @mathias */
